// @erfanmola/televue/dist/tele-vue-lib.js
import { provide as $t, openBlock as V, createElementBlock as tt, Fragment as Er, createElementVNode as J, normalizeClass as vt, unref as z, renderSlot as Rt, createBlock as Sr, Teleport as Ar, createCommentVNode as ye, ref as Ir, createTextVNode as Fr, inject as q, toDisplayString as Fe } from "vue";
import bt from "@twa-dev/sdk";
const Ur = {
  key: 0,
  href: "https://fonts.cdnfonts.com/css/sf-pro-display?styles=98774,98770",
  rel: "stylesheet"
}, Cr = {
  key: 1,
  href: "https://fonts.cdnfonts.com/css/roboto?styles=14394,14400",
  rel: "stylesheet"
};
var De, ke;
const Wi = {
  __name: "AppearanceProvider",
  props: {
    provideFonts: {
      default: !0,
      type: Boolean
    },
    platform: {
      default: (De = bt.platform) == null ? void 0 : De.toString().toLowerCase(),
      type: String
    },
    colorScheme: {
      default: (ke = bt.colorScheme) == null ? void 0 : ke.toString().toLowerCase(),
      type: String
    },
    theme: {
      default: null,
      type: String
    }
  },
  setup(o) {
    const u = o, f = u.platform === "ios" || u.platform === "macos" ? "apple" : "material";
    return $t("platform", u.platform), $t("colorScheme", u.colorScheme), $t("theme", u.theme || f), (l, h) => (V(), tt(Er, null, [
      J("div", {
        id: "tele-vue-appearance-provider",
        class: vt([`tele-vue-platform-${o.platform}`, `tele-vue-theme-${z(f)}`, `tele-vue-colorScheme-${o.colorScheme}`])
      }, [
        Rt(l.$slots, "default")
      ], 2),
      o.provideFonts ? (V(), Sr(Ar, {
        key: 0,
        to: "head"
      }, [
        (u.theme || z(f)) === "apple" ? (V(), tt("link", Ur)) : (V(), tt("link", Cr))
      ])) : ye("", !0)
    ], 64));
  }
}, qi = {
  __name: "LocaleProvider",
  props: {
    locale: {
      default: null,
      type: String
    },
    direction: {
      default: null,
      type: String
    }
  },
  setup(o) {
    var c, d;
    const u = o, f = [
      "ar",
      // Arabic
      "he",
      // Hebrew
      "fa",
      // Persian (Farsi)
      "ur",
      // Urdu
      "sd",
      // Sindhi
      "ps",
      // Pashto
      "yi"
      // Yiddish
    ], l = u.locale || ((d = (c = bt.initDataUnsafe.user) == null ? void 0 : c.language_code) == null ? void 0 : d.toString().toLowerCase()) || "en", h = u.direction || f.includes(l) ? "rtl" : "ltr";
    return document.body.classList.add(h), $t("locale", l), $t("direction", h), (i, B) => (V(), tt("div", {
      class: vt([
        `tele-vue-direction-${z(h)}`,
        `tele-vue-locale-${z(l)}`
      ])
    }, [
      Rt(i.$slots, "default")
    ], 2));
  }
};
var L = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function $r(o) {
  return o && o.__esModule && Object.prototype.hasOwnProperty.call(o, "default") ? o.default : o;
}
function Rr(o) {
  if (o.__esModule) return o;
  var u = o.default;
  if (typeof u == "function") {
    var f = function l() {
      return this instanceof l ? Reflect.construct(u, arguments, this.constructor) : u.apply(this, arguments);
    };
    f.prototype = u.prototype;
  } else f = {};
  return Object.defineProperty(f, "__esModule", { value: !0 }), Object.keys(o).forEach(function(l) {
    var h = Object.getOwnPropertyDescriptor(o, l);
    Object.defineProperty(f, l, h.get ? h : {
      enumerable: !0,
      get: function() {
        return o[l];
      }
    });
  }), f;
}
var Le = {}, Ne = {}, be = {};
Object.defineProperty(be, "__esModule", { value: !0 });
var Tr = {
  hashes: {
    "SHA3-512": { nodeName: "sha3-512", hashSize: 64, blockSize: 72 },
    "SHA3-384": { nodeName: "sha3-384", hashSize: 48, blockSize: 104 },
    "SHA3-256": { nodeName: "sha3-256", hashSize: 32, blockSize: 136 },
    "SHA3-224": { nodeName: "sha3-224", hashSize: 28, blockSize: 144 },
    "SHA-256": { nodeName: "sha256", hashSize: 32, blockSize: 64 },
    "SHA-384": { nodeName: "sha384", hashSize: 48, blockSize: 128 },
    "SHA-512": { nodeName: "sha512", hashSize: 64, blockSize: 128 },
    "SHA-1": { nodeName: "sha1", hashSize: 20, blockSize: 64 },
    MD5: { nodeName: "md5", hashSize: 16, blockSize: 64 }
  }
};
be.default = Tr;
var X = {};
const Mr = {}, Dr = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Mr
}, Symbol.toStringTag, { value: "Module" })), kr = /* @__PURE__ */ Rr(Dr);
Object.defineProperty(X, "__esModule", { value: !0 });
X.getCrypto = X.getRootWebCrypto = X.getWebCrypto = X.getNodeCrypto = void 0;
var _e = function() {
  if (!(typeof window < "u" && window.crypto))
    return typeof window > "u" && typeof crypto < "u" ? void 0 : kr;
};
X.getNodeCrypto = _e;
var ve = function() {
  if (typeof window < "u" && window.crypto)
    return window.crypto.subtle;
  if (typeof window > "u" && typeof crypto < "u")
    return crypto.subtle;
};
X.getWebCrypto = ve;
var Pe = function() {
  if (typeof window < "u" && window.crypto)
    return window.crypto;
  if (typeof window > "u" && typeof crypto < "u")
    return crypto;
};
X.getRootWebCrypto = Pe;
var He = function() {
  var o = ve(), u = _e();
  return typeof u < "u" ? { name: "nodeCrypto", crypto: u } : typeof o < "u" ? { name: "webCrypto", crypto: o } : { name: void 0 };
};
X.getCrypto = He;
X.default = { getNodeCrypto: _e, getWebCrypto: ve, getRootWebCrypto: Pe, getCrypto: He };
var Oe = {}, Zt = {}, Ee = {};
Object.defineProperty(Ee, "__esModule", { value: !0 });
var Lr = {
  hashes: {
    "SHA3-512": { nodeName: "sha3-512", hashSize: 64 },
    "SHA3-384": { nodeName: "sha3-384", hashSize: 48 },
    "SHA3-256": { nodeName: "sha3-256", hashSize: 32 },
    "SHA3-224": { nodeName: "sha3-224", hashSize: 28 },
    "SHA-256": { nodeName: "sha256", hashSize: 32 },
    "SHA-384": { nodeName: "sha384", hashSize: 48 },
    "SHA-512": { nodeName: "sha512", hashSize: 64 },
    "SHA-1": { nodeName: "sha1", hashSize: 20 },
    MD5: { nodeName: "md5", hashSize: 16 }
    // SHOULD NOT USE
  }
};
Ee.default = Lr;
var ze = { exports: {} }, We = { exports: {} };
(function() {
  var o = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/", u = {
    // Bit-wise rotation left
    rotl: function(f, l) {
      return f << l | f >>> 32 - l;
    },
    // Bit-wise rotation right
    rotr: function(f, l) {
      return f << 32 - l | f >>> l;
    },
    // Swap big-endian to little-endian and vice versa
    endian: function(f) {
      if (f.constructor == Number)
        return u.rotl(f, 8) & 16711935 | u.rotl(f, 24) & 4278255360;
      for (var l = 0; l < f.length; l++)
        f[l] = u.endian(f[l]);
      return f;
    },
    // Generate an array of any length of random bytes
    randomBytes: function(f) {
      for (var l = []; f > 0; f--)
        l.push(Math.floor(Math.random() * 256));
      return l;
    },
    // Convert a byte array to big-endian 32-bit words
    bytesToWords: function(f) {
      for (var l = [], h = 0, c = 0; h < f.length; h++, c += 8)
        l[c >>> 5] |= f[h] << 24 - c % 32;
      return l;
    },
    // Convert big-endian 32-bit words to a byte array
    wordsToBytes: function(f) {
      for (var l = [], h = 0; h < f.length * 32; h += 8)
        l.push(f[h >>> 5] >>> 24 - h % 32 & 255);
      return l;
    },
    // Convert a byte array to a hex string
    bytesToHex: function(f) {
      for (var l = [], h = 0; h < f.length; h++)
        l.push((f[h] >>> 4).toString(16)), l.push((f[h] & 15).toString(16));
      return l.join("");
    },
    // Convert a hex string to a byte array
    hexToBytes: function(f) {
      for (var l = [], h = 0; h < f.length; h += 2)
        l.push(parseInt(f.substr(h, 2), 16));
      return l;
    },
    // Convert a byte array to a base-64 string
    bytesToBase64: function(f) {
      for (var l = [], h = 0; h < f.length; h += 3)
        for (var c = f[h] << 16 | f[h + 1] << 8 | f[h + 2], d = 0; d < 4; d++)
          h * 8 + d * 6 <= f.length * 8 ? l.push(o.charAt(c >>> 6 * (3 - d) & 63)) : l.push("=");
      return l.join("");
    },
    // Convert a base-64 string to a byte array
    base64ToBytes: function(f) {
      f = f.replace(/[^A-Z0-9+\/]/ig, "");
      for (var l = [], h = 0, c = 0; h < f.length; c = ++h % 4)
        c != 0 && l.push((o.indexOf(f.charAt(h - 1)) & Math.pow(2, -2 * c + 8) - 1) << c * 2 | o.indexOf(f.charAt(h)) >>> 6 - c * 2);
      return l;
    }
  };
  We.exports = u;
})();
var Nr = We.exports, xe = {
  // UTF-8 encoding
  utf8: {
    // Convert a string to a byte array
    stringToBytes: function(o) {
      return xe.bin.stringToBytes(unescape(encodeURIComponent(o)));
    },
    // Convert a byte array to a string
    bytesToString: function(o) {
      return decodeURIComponent(escape(xe.bin.bytesToString(o)));
    }
  },
  // Binary encoding
  bin: {
    // Convert a string to a byte array
    stringToBytes: function(o) {
      for (var u = [], f = 0; f < o.length; f++)
        u.push(o.charCodeAt(f) & 255);
      return u;
    },
    // Convert a byte array to a string
    bytesToString: function(o) {
      for (var u = [], f = 0; f < o.length; f++)
        u.push(String.fromCharCode(o[f]));
      return u.join("");
    }
  }
}, Ue = xe;
/*!
 * Determine if an object is a Buffer
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
var Pr = function(o) {
  return o != null && (qe(o) || Hr(o) || !!o._isBuffer);
};
function qe(o) {
  return !!o.constructor && typeof o.constructor.isBuffer == "function" && o.constructor.isBuffer(o);
}
function Hr(o) {
  return typeof o.readFloatLE == "function" && typeof o.slice == "function" && qe(o.slice(0, 0));
}
(function() {
  var o = Nr, u = Ue.utf8, f = Pr, l = Ue.bin, h = function(c, d) {
    c.constructor == String ? d && d.encoding === "binary" ? c = l.stringToBytes(c) : c = u.stringToBytes(c) : f(c) ? c = Array.prototype.slice.call(c, 0) : !Array.isArray(c) && c.constructor !== Uint8Array && (c = c.toString());
    for (var i = o.bytesToWords(c), B = c.length * 8, p = 1732584193, x = -271733879, g = -1732584194, m = 271733878, w = 0; w < i.length; w++)
      i[w] = (i[w] << 8 | i[w] >>> 24) & 16711935 | (i[w] << 24 | i[w] >>> 8) & 4278255360;
    i[B >>> 5] |= 128 << B % 32, i[(B + 64 >>> 9 << 4) + 14] = B;
    for (var _ = h._ff, b = h._gg, v = h._hh, E = h._ii, w = 0; w < i.length; w += 16) {
      var S = p, I = x, M = g, U = m;
      p = _(p, x, g, m, i[w + 0], 7, -680876936), m = _(m, p, x, g, i[w + 1], 12, -389564586), g = _(g, m, p, x, i[w + 2], 17, 606105819), x = _(x, g, m, p, i[w + 3], 22, -1044525330), p = _(p, x, g, m, i[w + 4], 7, -176418897), m = _(m, p, x, g, i[w + 5], 12, 1200080426), g = _(g, m, p, x, i[w + 6], 17, -1473231341), x = _(x, g, m, p, i[w + 7], 22, -45705983), p = _(p, x, g, m, i[w + 8], 7, 1770035416), m = _(m, p, x, g, i[w + 9], 12, -1958414417), g = _(g, m, p, x, i[w + 10], 17, -42063), x = _(x, g, m, p, i[w + 11], 22, -1990404162), p = _(p, x, g, m, i[w + 12], 7, 1804603682), m = _(m, p, x, g, i[w + 13], 12, -40341101), g = _(g, m, p, x, i[w + 14], 17, -1502002290), x = _(x, g, m, p, i[w + 15], 22, 1236535329), p = b(p, x, g, m, i[w + 1], 5, -165796510), m = b(m, p, x, g, i[w + 6], 9, -1069501632), g = b(g, m, p, x, i[w + 11], 14, 643717713), x = b(x, g, m, p, i[w + 0], 20, -373897302), p = b(p, x, g, m, i[w + 5], 5, -701558691), m = b(m, p, x, g, i[w + 10], 9, 38016083), g = b(g, m, p, x, i[w + 15], 14, -660478335), x = b(x, g, m, p, i[w + 4], 20, -405537848), p = b(p, x, g, m, i[w + 9], 5, 568446438), m = b(m, p, x, g, i[w + 14], 9, -1019803690), g = b(g, m, p, x, i[w + 3], 14, -187363961), x = b(x, g, m, p, i[w + 8], 20, 1163531501), p = b(p, x, g, m, i[w + 13], 5, -1444681467), m = b(m, p, x, g, i[w + 2], 9, -51403784), g = b(g, m, p, x, i[w + 7], 14, 1735328473), x = b(x, g, m, p, i[w + 12], 20, -1926607734), p = v(p, x, g, m, i[w + 5], 4, -378558), m = v(m, p, x, g, i[w + 8], 11, -2022574463), g = v(g, m, p, x, i[w + 11], 16, 1839030562), x = v(x, g, m, p, i[w + 14], 23, -35309556), p = v(p, x, g, m, i[w + 1], 4, -1530992060), m = v(m, p, x, g, i[w + 4], 11, 1272893353), g = v(g, m, p, x, i[w + 7], 16, -155497632), x = v(x, g, m, p, i[w + 10], 23, -1094730640), p = v(p, x, g, m, i[w + 13], 4, 681279174), m = v(m, p, x, g, i[w + 0], 11, -358537222), g = v(g, m, p, x, i[w + 3], 16, -722521979), x = v(x, g, m, p, i[w + 6], 23, 76029189), p = v(p, x, g, m, i[w + 9], 4, -640364487), m = v(m, p, x, g, i[w + 12], 11, -421815835), g = v(g, m, p, x, i[w + 15], 16, 530742520), x = v(x, g, m, p, i[w + 2], 23, -995338651), p = E(p, x, g, m, i[w + 0], 6, -198630844), m = E(m, p, x, g, i[w + 7], 10, 1126891415), g = E(g, m, p, x, i[w + 14], 15, -1416354905), x = E(x, g, m, p, i[w + 5], 21, -57434055), p = E(p, x, g, m, i[w + 12], 6, 1700485571), m = E(m, p, x, g, i[w + 3], 10, -1894986606), g = E(g, m, p, x, i[w + 10], 15, -1051523), x = E(x, g, m, p, i[w + 1], 21, -2054922799), p = E(p, x, g, m, i[w + 8], 6, 1873313359), m = E(m, p, x, g, i[w + 15], 10, -30611744), g = E(g, m, p, x, i[w + 6], 15, -1560198380), x = E(x, g, m, p, i[w + 13], 21, 1309151649), p = E(p, x, g, m, i[w + 4], 6, -145523070), m = E(m, p, x, g, i[w + 11], 10, -1120210379), g = E(g, m, p, x, i[w + 2], 15, 718787259), x = E(x, g, m, p, i[w + 9], 21, -343485551), p = p + S >>> 0, x = x + I >>> 0, g = g + M >>> 0, m = m + U >>> 0;
    }
    return o.endian([p, x, g, m]);
  };
  h._ff = function(c, d, i, B, p, x, g) {
    var m = c + (d & i | ~d & B) + (p >>> 0) + g;
    return (m << x | m >>> 32 - x) + d;
  }, h._gg = function(c, d, i, B, p, x, g) {
    var m = c + (d & B | i & ~B) + (p >>> 0) + g;
    return (m << x | m >>> 32 - x) + d;
  }, h._hh = function(c, d, i, B, p, x, g) {
    var m = c + (d ^ i ^ B) + (p >>> 0) + g;
    return (m << x | m >>> 32 - x) + d;
  }, h._ii = function(c, d, i, B, p, x, g) {
    var m = c + (i ^ (d | ~B)) + (p >>> 0) + g;
    return (m << x | m >>> 32 - x) + d;
  }, h._blocksize = 16, h._digestsize = 16, ze.exports = function(c, d) {
    if (c == null)
      throw new Error("Illegal argument " + c);
    var i = o.wordsToBytes(h(c, d));
    return d && d.asBytes ? i : d && d.asString ? l.bytesToString(i) : o.bytesToHex(i);
  };
})();
var Or = ze.exports, je = {}, Se = {}, Vt = {};
Vt.byteLength = qr;
Vt.toByteArray = Gr;
Vt.fromByteArray = Vr;
var it = [], Y = [], zr = typeof Uint8Array < "u" ? Uint8Array : Array, he = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
for (var Ct = 0, Wr = he.length; Ct < Wr; ++Ct)
  it[Ct] = he[Ct], Y[he.charCodeAt(Ct)] = Ct;
Y[45] = 62;
Y[95] = 63;
function Ge(o) {
  var u = o.length;
  if (u % 4 > 0)
    throw new Error("Invalid string. Length must be a multiple of 4");
  var f = o.indexOf("=");
  f === -1 && (f = u);
  var l = f === u ? 0 : 4 - f % 4;
  return [f, l];
}
function qr(o) {
  var u = Ge(o), f = u[0], l = u[1];
  return (f + l) * 3 / 4 - l;
}
function jr(o, u, f) {
  return (u + f) * 3 / 4 - f;
}
function Gr(o) {
  var u, f = Ge(o), l = f[0], h = f[1], c = new zr(jr(o, l, h)), d = 0, i = h > 0 ? l - 4 : l, B;
  for (B = 0; B < i; B += 4)
    u = Y[o.charCodeAt(B)] << 18 | Y[o.charCodeAt(B + 1)] << 12 | Y[o.charCodeAt(B + 2)] << 6 | Y[o.charCodeAt(B + 3)], c[d++] = u >> 16 & 255, c[d++] = u >> 8 & 255, c[d++] = u & 255;
  return h === 2 && (u = Y[o.charCodeAt(B)] << 2 | Y[o.charCodeAt(B + 1)] >> 4, c[d++] = u & 255), h === 1 && (u = Y[o.charCodeAt(B)] << 10 | Y[o.charCodeAt(B + 1)] << 4 | Y[o.charCodeAt(B + 2)] >> 2, c[d++] = u >> 8 & 255, c[d++] = u & 255), c;
}
function Kr(o) {
  return it[o >> 18 & 63] + it[o >> 12 & 63] + it[o >> 6 & 63] + it[o & 63];
}
function Yr(o, u, f) {
  for (var l, h = [], c = u; c < f; c += 3)
    l = (o[c] << 16 & 16711680) + (o[c + 1] << 8 & 65280) + (o[c + 2] & 255), h.push(Kr(l));
  return h.join("");
}
function Vr(o) {
  for (var u, f = o.length, l = f % 3, h = [], c = 16383, d = 0, i = f - l; d < i; d += c)
    h.push(Yr(o, d, d + c > i ? i : d + c));
  return l === 1 ? (u = o[f - 1], h.push(
    it[u >> 2] + it[u << 4 & 63] + "=="
  )) : l === 2 && (u = (o[f - 2] << 8) + o[f - 1], h.push(
    it[u >> 10] + it[u >> 4 & 63] + it[u << 2 & 63] + "="
  )), h.join("");
}
var Qt = {};
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
Qt.read = function(o, u, f, l, h) {
  var c, d, i = h * 8 - l - 1, B = (1 << i) - 1, p = B >> 1, x = -7, g = f ? h - 1 : 0, m = f ? -1 : 1, w = o[u + g];
  for (g += m, c = w & (1 << -x) - 1, w >>= -x, x += i; x > 0; c = c * 256 + o[u + g], g += m, x -= 8)
    ;
  for (d = c & (1 << -x) - 1, c >>= -x, x += l; x > 0; d = d * 256 + o[u + g], g += m, x -= 8)
    ;
  if (c === 0)
    c = 1 - p;
  else {
    if (c === B)
      return d ? NaN : (w ? -1 : 1) * (1 / 0);
    d = d + Math.pow(2, l), c = c - p;
  }
  return (w ? -1 : 1) * d * Math.pow(2, c - l);
};
Qt.write = function(o, u, f, l, h, c) {
  var d, i, B, p = c * 8 - h - 1, x = (1 << p) - 1, g = x >> 1, m = h === 23 ? Math.pow(2, -24) - Math.pow(2, -77) : 0, w = l ? 0 : c - 1, _ = l ? 1 : -1, b = u < 0 || u === 0 && 1 / u < 0 ? 1 : 0;
  for (u = Math.abs(u), isNaN(u) || u === 1 / 0 ? (i = isNaN(u) ? 1 : 0, d = x) : (d = Math.floor(Math.log(u) / Math.LN2), u * (B = Math.pow(2, -d)) < 1 && (d--, B *= 2), d + g >= 1 ? u += m / B : u += m * Math.pow(2, 1 - g), u * B >= 2 && (d++, B /= 2), d + g >= x ? (i = 0, d = x) : d + g >= 1 ? (i = (u * B - 1) * Math.pow(2, h), d = d + g) : (i = u * Math.pow(2, g - 1) * Math.pow(2, h), d = 0)); h >= 8; o[f + w] = i & 255, w += _, i /= 256, h -= 8)
    ;
  for (d = d << h | i, p += h; p > 0; o[f + w] = d & 255, w += _, d /= 256, p -= 8)
    ;
  o[f + w - _] |= b * 128;
};
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
(function(o) {
  const u = Vt, f = Qt, l = typeof Symbol == "function" && typeof Symbol.for == "function" ? Symbol.for("nodejs.util.inspect.custom") : null;
  o.Buffer = i, o.SlowBuffer = S, o.INSPECT_MAX_BYTES = 50;
  const h = 2147483647;
  o.kMaxLength = h, i.TYPED_ARRAY_SUPPORT = c(), !i.TYPED_ARRAY_SUPPORT && typeof console < "u" && typeof console.error == "function" && console.error(
    "This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support."
  );
  function c() {
    try {
      const r = new Uint8Array(1), t = { foo: function() {
        return 42;
      } };
      return Object.setPrototypeOf(t, Uint8Array.prototype), Object.setPrototypeOf(r, t), r.foo() === 42;
    } catch {
      return !1;
    }
  }
  Object.defineProperty(i.prototype, "parent", {
    enumerable: !0,
    get: function() {
      if (i.isBuffer(this))
        return this.buffer;
    }
  }), Object.defineProperty(i.prototype, "offset", {
    enumerable: !0,
    get: function() {
      if (i.isBuffer(this))
        return this.byteOffset;
    }
  });
  function d(r) {
    if (r > h)
      throw new RangeError('The value "' + r + '" is invalid for option "size"');
    const t = new Uint8Array(r);
    return Object.setPrototypeOf(t, i.prototype), t;
  }
  function i(r, t, e) {
    if (typeof r == "number") {
      if (typeof t == "string")
        throw new TypeError(
          'The "string" argument must be of type string. Received type number'
        );
      return g(r);
    }
    return B(r, t, e);
  }
  i.poolSize = 8192;
  function B(r, t, e) {
    if (typeof r == "string")
      return m(r, t);
    if (ArrayBuffer.isView(r))
      return _(r);
    if (r == null)
      throw new TypeError(
        "The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type " + typeof r
      );
    if (O(r, ArrayBuffer) || r && O(r.buffer, ArrayBuffer) || typeof SharedArrayBuffer < "u" && (O(r, SharedArrayBuffer) || r && O(r.buffer, SharedArrayBuffer)))
      return b(r, t, e);
    if (typeof r == "number")
      throw new TypeError(
        'The "value" argument must not be of type number. Received type number'
      );
    const n = r.valueOf && r.valueOf();
    if (n != null && n !== r)
      return i.from(n, t, e);
    const a = v(r);
    if (a) return a;
    if (typeof Symbol < "u" && Symbol.toPrimitive != null && typeof r[Symbol.toPrimitive] == "function")
      return i.from(r[Symbol.toPrimitive]("string"), t, e);
    throw new TypeError(
      "The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type " + typeof r
    );
  }
  i.from = function(r, t, e) {
    return B(r, t, e);
  }, Object.setPrototypeOf(i.prototype, Uint8Array.prototype), Object.setPrototypeOf(i, Uint8Array);
  function p(r) {
    if (typeof r != "number")
      throw new TypeError('"size" argument must be of type number');
    if (r < 0)
      throw new RangeError('The value "' + r + '" is invalid for option "size"');
  }
  function x(r, t, e) {
    return p(r), r <= 0 ? d(r) : t !== void 0 ? typeof e == "string" ? d(r).fill(t, e) : d(r).fill(t) : d(r);
  }
  i.alloc = function(r, t, e) {
    return x(r, t, e);
  };
  function g(r) {
    return p(r), d(r < 0 ? 0 : E(r) | 0);
  }
  i.allocUnsafe = function(r) {
    return g(r);
  }, i.allocUnsafeSlow = function(r) {
    return g(r);
  };
  function m(r, t) {
    if ((typeof t != "string" || t === "") && (t = "utf8"), !i.isEncoding(t))
      throw new TypeError("Unknown encoding: " + t);
    const e = I(r, t) | 0;
    let n = d(e);
    const a = n.write(r, t);
    return a !== e && (n = n.slice(0, a)), n;
  }
  function w(r) {
    const t = r.length < 0 ? 0 : E(r.length) | 0, e = d(t);
    for (let n = 0; n < t; n += 1)
      e[n] = r[n] & 255;
    return e;
  }
  function _(r) {
    if (O(r, Uint8Array)) {
      const t = new Uint8Array(r);
      return b(t.buffer, t.byteOffset, t.byteLength);
    }
    return w(r);
  }
  function b(r, t, e) {
    if (t < 0 || r.byteLength < t)
      throw new RangeError('"offset" is outside of buffer bounds');
    if (r.byteLength < t + (e || 0))
      throw new RangeError('"length" is outside of buffer bounds');
    let n;
    return t === void 0 && e === void 0 ? n = new Uint8Array(r) : e === void 0 ? n = new Uint8Array(r, t) : n = new Uint8Array(r, t, e), Object.setPrototypeOf(n, i.prototype), n;
  }
  function v(r) {
    if (i.isBuffer(r)) {
      const t = E(r.length) | 0, e = d(t);
      return e.length === 0 || r.copy(e, 0, 0, t), e;
    }
    if (r.length !== void 0)
      return typeof r.length != "number" || Bt(r.length) ? d(0) : w(r);
    if (r.type === "Buffer" && Array.isArray(r.data))
      return w(r.data);
  }
  function E(r) {
    if (r >= h)
      throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x" + h.toString(16) + " bytes");
    return r | 0;
  }
  function S(r) {
    return +r != r && (r = 0), i.alloc(+r);
  }
  i.isBuffer = function(t) {
    return t != null && t._isBuffer === !0 && t !== i.prototype;
  }, i.compare = function(t, e) {
    if (O(t, Uint8Array) && (t = i.from(t, t.offset, t.byteLength)), O(e, Uint8Array) && (e = i.from(e, e.offset, e.byteLength)), !i.isBuffer(t) || !i.isBuffer(e))
      throw new TypeError(
        'The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array'
      );
    if (t === e) return 0;
    let n = t.length, a = e.length;
    for (let s = 0, y = Math.min(n, a); s < y; ++s)
      if (t[s] !== e[s]) {
        n = t[s], a = e[s];
        break;
      }
    return n < a ? -1 : a < n ? 1 : 0;
  }, i.isEncoding = function(t) {
    switch (String(t).toLowerCase()) {
      case "hex":
      case "utf8":
      case "utf-8":
      case "ascii":
      case "latin1":
      case "binary":
      case "base64":
      case "ucs2":
      case "ucs-2":
      case "utf16le":
      case "utf-16le":
        return !0;
      default:
        return !1;
    }
  }, i.concat = function(t, e) {
    if (!Array.isArray(t))
      throw new TypeError('"list" argument must be an Array of Buffers');
    if (t.length === 0)
      return i.alloc(0);
    let n;
    if (e === void 0)
      for (e = 0, n = 0; n < t.length; ++n)
        e += t[n].length;
    const a = i.allocUnsafe(e);
    let s = 0;
    for (n = 0; n < t.length; ++n) {
      let y = t[n];
      if (O(y, Uint8Array))
        s + y.length > a.length ? (i.isBuffer(y) || (y = i.from(y)), y.copy(a, s)) : Uint8Array.prototype.set.call(
          a,
          y,
          s
        );
      else if (i.isBuffer(y))
        y.copy(a, s);
      else
        throw new TypeError('"list" argument must be an Array of Buffers');
      s += y.length;
    }
    return a;
  };
  function I(r, t) {
    if (i.isBuffer(r))
      return r.length;
    if (ArrayBuffer.isView(r) || O(r, ArrayBuffer))
      return r.byteLength;
    if (typeof r != "string")
      throw new TypeError(
        'The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type ' + typeof r
      );
    const e = r.length, n = arguments.length > 2 && arguments[2] === !0;
    if (!n && e === 0) return 0;
    let a = !1;
    for (; ; )
      switch (t) {
        case "ascii":
        case "latin1":
        case "binary":
          return e;
        case "utf8":
        case "utf-8":
          return mt(r).length;
        case "ucs2":
        case "ucs-2":
        case "utf16le":
        case "utf-16le":
          return e * 2;
        case "hex":
          return e >>> 1;
        case "base64":
          return Gt(r).length;
        default:
          if (a)
            return n ? -1 : mt(r).length;
          t = ("" + t).toLowerCase(), a = !0;
      }
  }
  i.byteLength = I;
  function M(r, t, e) {
    let n = !1;
    if ((t === void 0 || t < 0) && (t = 0), t > this.length || ((e === void 0 || e > this.length) && (e = this.length), e <= 0) || (e >>>= 0, t >>>= 0, e <= t))
      return "";
    for (r || (r = "utf8"); ; )
      switch (r) {
        case "hex":
          return ee(this, t, e);
        case "utf8":
        case "utf-8":
          return yt(this, t, e);
        case "ascii":
          return Ft(this, t, e);
        case "latin1":
        case "binary":
          return Ut(this, t, e);
        case "base64":
          return It(this, t, e);
        case "ucs2":
        case "ucs-2":
        case "utf16le":
        case "utf-16le":
          return re(this, t, e);
        default:
          if (n) throw new TypeError("Unknown encoding: " + r);
          r = (r + "").toLowerCase(), n = !0;
      }
  }
  i.prototype._isBuffer = !0;
  function U(r, t, e) {
    const n = r[t];
    r[t] = r[e], r[e] = n;
  }
  i.prototype.swap16 = function() {
    const t = this.length;
    if (t % 2 !== 0)
      throw new RangeError("Buffer size must be a multiple of 16-bits");
    for (let e = 0; e < t; e += 2)
      U(this, e, e + 1);
    return this;
  }, i.prototype.swap32 = function() {
    const t = this.length;
    if (t % 4 !== 0)
      throw new RangeError("Buffer size must be a multiple of 32-bits");
    for (let e = 0; e < t; e += 4)
      U(this, e, e + 3), U(this, e + 1, e + 2);
    return this;
  }, i.prototype.swap64 = function() {
    const t = this.length;
    if (t % 8 !== 0)
      throw new RangeError("Buffer size must be a multiple of 64-bits");
    for (let e = 0; e < t; e += 8)
      U(this, e, e + 7), U(this, e + 1, e + 6), U(this, e + 2, e + 5), U(this, e + 3, e + 4);
    return this;
  }, i.prototype.toString = function() {
    const t = this.length;
    return t === 0 ? "" : arguments.length === 0 ? yt(this, 0, t) : M.apply(this, arguments);
  }, i.prototype.toLocaleString = i.prototype.toString, i.prototype.equals = function(t) {
    if (!i.isBuffer(t)) throw new TypeError("Argument must be a Buffer");
    return this === t ? !0 : i.compare(this, t) === 0;
  }, i.prototype.inspect = function() {
    let t = "";
    const e = o.INSPECT_MAX_BYTES;
    return t = this.toString("hex", 0, e).replace(/(.{2})/g, "$1 ").trim(), this.length > e && (t += " ... "), "<Buffer " + t + ">";
  }, l && (i.prototype[l] = i.prototype.inspect), i.prototype.compare = function(t, e, n, a, s) {
    if (O(t, Uint8Array) && (t = i.from(t, t.offset, t.byteLength)), !i.isBuffer(t))
      throw new TypeError(
        'The "target" argument must be one of type Buffer or Uint8Array. Received type ' + typeof t
      );
    if (e === void 0 && (e = 0), n === void 0 && (n = t ? t.length : 0), a === void 0 && (a = 0), s === void 0 && (s = this.length), e < 0 || n > t.length || a < 0 || s > this.length)
      throw new RangeError("out of range index");
    if (a >= s && e >= n)
      return 0;
    if (a >= s)
      return -1;
    if (e >= n)
      return 1;
    if (e >>>= 0, n >>>= 0, a >>>= 0, s >>>= 0, this === t) return 0;
    let y = s - a, A = n - e;
    const R = Math.min(y, A), $ = this.slice(a, s), T = t.slice(e, n);
    for (let C = 0; C < R; ++C)
      if ($[C] !== T[C]) {
        y = $[C], A = T[C];
        break;
      }
    return y < A ? -1 : A < y ? 1 : 0;
  };
  function F(r, t, e, n, a) {
    if (r.length === 0) return -1;
    if (typeof e == "string" ? (n = e, e = 0) : e > 2147483647 ? e = 2147483647 : e < -2147483648 && (e = -2147483648), e = +e, Bt(e) && (e = a ? 0 : r.length - 1), e < 0 && (e = r.length + e), e >= r.length) {
      if (a) return -1;
      e = r.length - 1;
    } else if (e < 0)
      if (a) e = 0;
      else return -1;
    if (typeof t == "string" && (t = i.from(t, n)), i.isBuffer(t))
      return t.length === 0 ? -1 : D(r, t, e, n, a);
    if (typeof t == "number")
      return t = t & 255, typeof Uint8Array.prototype.indexOf == "function" ? a ? Uint8Array.prototype.indexOf.call(r, t, e) : Uint8Array.prototype.lastIndexOf.call(r, t, e) : D(r, [t], e, n, a);
    throw new TypeError("val must be string, number or Buffer");
  }
  function D(r, t, e, n, a) {
    let s = 1, y = r.length, A = t.length;
    if (n !== void 0 && (n = String(n).toLowerCase(), n === "ucs2" || n === "ucs-2" || n === "utf16le" || n === "utf-16le")) {
      if (r.length < 2 || t.length < 2)
        return -1;
      s = 2, y /= 2, A /= 2, e /= 2;
    }
    function R(T, C) {
      return s === 1 ? T[C] : T.readUInt16BE(C * s);
    }
    let $;
    if (a) {
      let T = -1;
      for ($ = e; $ < y; $++)
        if (R(r, $) === R(t, T === -1 ? 0 : $ - T)) {
          if (T === -1 && (T = $), $ - T + 1 === A) return T * s;
        } else
          T !== -1 && ($ -= $ - T), T = -1;
    } else
      for (e + A > y && (e = y - A), $ = e; $ >= 0; $--) {
        let T = !0;
        for (let C = 0; C < A; C++)
          if (R(r, $ + C) !== R(t, C)) {
            T = !1;
            break;
          }
        if (T) return $;
      }
    return -1;
  }
  i.prototype.includes = function(t, e, n) {
    return this.indexOf(t, e, n) !== -1;
  }, i.prototype.indexOf = function(t, e, n) {
    return F(this, t, e, n, !0);
  }, i.prototype.lastIndexOf = function(t, e, n) {
    return F(this, t, e, n, !1);
  };
  function P(r, t, e, n) {
    e = Number(e) || 0;
    const a = r.length - e;
    n ? (n = Number(n), n > a && (n = a)) : n = a;
    const s = t.length;
    n > s / 2 && (n = s / 2);
    let y;
    for (y = 0; y < n; ++y) {
      const A = parseInt(t.substr(y * 2, 2), 16);
      if (Bt(A)) return y;
      r[e + y] = A;
    }
    return y;
  }
  function rt(r, t, e, n) {
    return lt(mt(t, r.length - e), r, e, n);
  }
  function Et(r, t, e, n) {
    return lt(ue(t), r, e, n);
  }
  function St(r, t, e, n) {
    return lt(Gt(t), r, e, n);
  }
  function At(r, t, e, n) {
    return lt(ae(t, r.length - e), r, e, n);
  }
  i.prototype.write = function(t, e, n, a) {
    if (e === void 0)
      a = "utf8", n = this.length, e = 0;
    else if (n === void 0 && typeof e == "string")
      a = e, n = this.length, e = 0;
    else if (isFinite(e))
      e = e >>> 0, isFinite(n) ? (n = n >>> 0, a === void 0 && (a = "utf8")) : (a = n, n = void 0);
    else
      throw new Error(
        "Buffer.write(string, encoding, offset[, length]) is no longer supported"
      );
    const s = this.length - e;
    if ((n === void 0 || n > s) && (n = s), t.length > 0 && (n < 0 || e < 0) || e > this.length)
      throw new RangeError("Attempt to write outside buffer bounds");
    a || (a = "utf8");
    let y = !1;
    for (; ; )
      switch (a) {
        case "hex":
          return P(this, t, e, n);
        case "utf8":
        case "utf-8":
          return rt(this, t, e, n);
        case "ascii":
        case "latin1":
        case "binary":
          return Et(this, t, e, n);
        case "base64":
          return St(this, t, e, n);
        case "ucs2":
        case "ucs-2":
        case "utf16le":
        case "utf-16le":
          return At(this, t, e, n);
        default:
          if (y) throw new TypeError("Unknown encoding: " + a);
          a = ("" + a).toLowerCase(), y = !0;
      }
  }, i.prototype.toJSON = function() {
    return {
      type: "Buffer",
      data: Array.prototype.slice.call(this._arr || this, 0)
    };
  };
  function It(r, t, e) {
    return t === 0 && e === r.length ? u.fromByteArray(r) : u.fromByteArray(r.slice(t, e));
  }
  function yt(r, t, e) {
    e = Math.min(r.length, e);
    const n = [];
    let a = t;
    for (; a < e; ) {
      const s = r[a];
      let y = null, A = s > 239 ? 4 : s > 223 ? 3 : s > 191 ? 2 : 1;
      if (a + A <= e) {
        let R, $, T, C;
        switch (A) {
          case 1:
            s < 128 && (y = s);
            break;
          case 2:
            R = r[a + 1], (R & 192) === 128 && (C = (s & 31) << 6 | R & 63, C > 127 && (y = C));
            break;
          case 3:
            R = r[a + 1], $ = r[a + 2], (R & 192) === 128 && ($ & 192) === 128 && (C = (s & 15) << 12 | (R & 63) << 6 | $ & 63, C > 2047 && (C < 55296 || C > 57343) && (y = C));
            break;
          case 4:
            R = r[a + 1], $ = r[a + 2], T = r[a + 3], (R & 192) === 128 && ($ & 192) === 128 && (T & 192) === 128 && (C = (s & 15) << 18 | (R & 63) << 12 | ($ & 63) << 6 | T & 63, C > 65535 && C < 1114112 && (y = C));
        }
      }
      y === null ? (y = 65533, A = 1) : y > 65535 && (y -= 65536, n.push(y >>> 10 & 1023 | 55296), y = 56320 | y & 1023), n.push(y), a += A;
    }
    return xt(n);
  }
  const nt = 4096;
  function xt(r) {
    const t = r.length;
    if (t <= nt)
      return String.fromCharCode.apply(String, r);
    let e = "", n = 0;
    for (; n < t; )
      e += String.fromCharCode.apply(
        String,
        r.slice(n, n += nt)
      );
    return e;
  }
  function Ft(r, t, e) {
    let n = "";
    e = Math.min(r.length, e);
    for (let a = t; a < e; ++a)
      n += String.fromCharCode(r[a] & 127);
    return n;
  }
  function Ut(r, t, e) {
    let n = "";
    e = Math.min(r.length, e);
    for (let a = t; a < e; ++a)
      n += String.fromCharCode(r[a]);
    return n;
  }
  function ee(r, t, e) {
    const n = r.length;
    (!t || t < 0) && (t = 0), (!e || e < 0 || e > n) && (e = n);
    let a = "";
    for (let s = t; s < e; ++s)
      a += fe[r[s]];
    return a;
  }
  function re(r, t, e) {
    const n = r.slice(t, e);
    let a = "";
    for (let s = 0; s < n.length - 1; s += 2)
      a += String.fromCharCode(n[s] + n[s + 1] * 256);
    return a;
  }
  i.prototype.slice = function(t, e) {
    const n = this.length;
    t = ~~t, e = e === void 0 ? n : ~~e, t < 0 ? (t += n, t < 0 && (t = 0)) : t > n && (t = n), e < 0 ? (e += n, e < 0 && (e = 0)) : e > n && (e = n), e < t && (e = t);
    const a = this.subarray(t, e);
    return Object.setPrototypeOf(a, i.prototype), a;
  };
  function k(r, t, e) {
    if (r % 1 !== 0 || r < 0) throw new RangeError("offset is not uint");
    if (r + t > e) throw new RangeError("Trying to access beyond buffer length");
  }
  i.prototype.readUintLE = i.prototype.readUIntLE = function(t, e, n) {
    t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
    let a = this[t], s = 1, y = 0;
    for (; ++y < e && (s *= 256); )
      a += this[t + y] * s;
    return a;
  }, i.prototype.readUintBE = i.prototype.readUIntBE = function(t, e, n) {
    t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
    let a = this[t + --e], s = 1;
    for (; e > 0 && (s *= 256); )
      a += this[t + --e] * s;
    return a;
  }, i.prototype.readUint8 = i.prototype.readUInt8 = function(t, e) {
    return t = t >>> 0, e || k(t, 1, this.length), this[t];
  }, i.prototype.readUint16LE = i.prototype.readUInt16LE = function(t, e) {
    return t = t >>> 0, e || k(t, 2, this.length), this[t] | this[t + 1] << 8;
  }, i.prototype.readUint16BE = i.prototype.readUInt16BE = function(t, e) {
    return t = t >>> 0, e || k(t, 2, this.length), this[t] << 8 | this[t + 1];
  }, i.prototype.readUint32LE = i.prototype.readUInt32LE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), (this[t] | this[t + 1] << 8 | this[t + 2] << 16) + this[t + 3] * 16777216;
  }, i.prototype.readUint32BE = i.prototype.readUInt32BE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), this[t] * 16777216 + (this[t + 1] << 16 | this[t + 2] << 8 | this[t + 3]);
  }, i.prototype.readBigUInt64LE = W(function(t) {
    t = t >>> 0, K(t, "offset");
    const e = this[t], n = this[t + 7];
    (e === void 0 || n === void 0) && Z(t, this.length - 8);
    const a = e + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + this[++t] * 2 ** 24, s = this[++t] + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + n * 2 ** 24;
    return BigInt(a) + (BigInt(s) << BigInt(32));
  }), i.prototype.readBigUInt64BE = W(function(t) {
    t = t >>> 0, K(t, "offset");
    const e = this[t], n = this[t + 7];
    (e === void 0 || n === void 0) && Z(t, this.length - 8);
    const a = e * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + this[++t], s = this[++t] * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + n;
    return (BigInt(a) << BigInt(32)) + BigInt(s);
  }), i.prototype.readIntLE = function(t, e, n) {
    t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
    let a = this[t], s = 1, y = 0;
    for (; ++y < e && (s *= 256); )
      a += this[t + y] * s;
    return s *= 128, a >= s && (a -= Math.pow(2, 8 * e)), a;
  }, i.prototype.readIntBE = function(t, e, n) {
    t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
    let a = e, s = 1, y = this[t + --a];
    for (; a > 0 && (s *= 256); )
      y += this[t + --a] * s;
    return s *= 128, y >= s && (y -= Math.pow(2, 8 * e)), y;
  }, i.prototype.readInt8 = function(t, e) {
    return t = t >>> 0, e || k(t, 1, this.length), this[t] & 128 ? (255 - this[t] + 1) * -1 : this[t];
  }, i.prototype.readInt16LE = function(t, e) {
    t = t >>> 0, e || k(t, 2, this.length);
    const n = this[t] | this[t + 1] << 8;
    return n & 32768 ? n | 4294901760 : n;
  }, i.prototype.readInt16BE = function(t, e) {
    t = t >>> 0, e || k(t, 2, this.length);
    const n = this[t + 1] | this[t] << 8;
    return n & 32768 ? n | 4294901760 : n;
  }, i.prototype.readInt32LE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), this[t] | this[t + 1] << 8 | this[t + 2] << 16 | this[t + 3] << 24;
  }, i.prototype.readInt32BE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), this[t] << 24 | this[t + 1] << 16 | this[t + 2] << 8 | this[t + 3];
  }, i.prototype.readBigInt64LE = W(function(t) {
    t = t >>> 0, K(t, "offset");
    const e = this[t], n = this[t + 7];
    (e === void 0 || n === void 0) && Z(t, this.length - 8);
    const a = this[t + 4] + this[t + 5] * 2 ** 8 + this[t + 6] * 2 ** 16 + (n << 24);
    return (BigInt(a) << BigInt(32)) + BigInt(e + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + this[++t] * 2 ** 24);
  }), i.prototype.readBigInt64BE = W(function(t) {
    t = t >>> 0, K(t, "offset");
    const e = this[t], n = this[t + 7];
    (e === void 0 || n === void 0) && Z(t, this.length - 8);
    const a = (e << 24) + // Overflow
    this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + this[++t];
    return (BigInt(a) << BigInt(32)) + BigInt(this[++t] * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + n);
  }), i.prototype.readFloatLE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), f.read(this, t, !0, 23, 4);
  }, i.prototype.readFloatBE = function(t, e) {
    return t = t >>> 0, e || k(t, 4, this.length), f.read(this, t, !1, 23, 4);
  }, i.prototype.readDoubleLE = function(t, e) {
    return t = t >>> 0, e || k(t, 8, this.length), f.read(this, t, !0, 52, 8);
  }, i.prototype.readDoubleBE = function(t, e) {
    return t = t >>> 0, e || k(t, 8, this.length), f.read(this, t, !1, 52, 8);
  };
  function H(r, t, e, n, a, s) {
    if (!i.isBuffer(r)) throw new TypeError('"buffer" argument must be a Buffer instance');
    if (t > a || t < s) throw new RangeError('"value" argument is out of bounds');
    if (e + n > r.length) throw new RangeError("Index out of range");
  }
  i.prototype.writeUintLE = i.prototype.writeUIntLE = function(t, e, n, a) {
    if (t = +t, e = e >>> 0, n = n >>> 0, !a) {
      const A = Math.pow(2, 8 * n) - 1;
      H(this, t, e, n, A, 0);
    }
    let s = 1, y = 0;
    for (this[e] = t & 255; ++y < n && (s *= 256); )
      this[e + y] = t / s & 255;
    return e + n;
  }, i.prototype.writeUintBE = i.prototype.writeUIntBE = function(t, e, n, a) {
    if (t = +t, e = e >>> 0, n = n >>> 0, !a) {
      const A = Math.pow(2, 8 * n) - 1;
      H(this, t, e, n, A, 0);
    }
    let s = n - 1, y = 1;
    for (this[e + s] = t & 255; --s >= 0 && (y *= 256); )
      this[e + s] = t / y & 255;
    return e + n;
  }, i.prototype.writeUint8 = i.prototype.writeUInt8 = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 1, 255, 0), this[e] = t & 255, e + 1;
  }, i.prototype.writeUint16LE = i.prototype.writeUInt16LE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 2, 65535, 0), this[e] = t & 255, this[e + 1] = t >>> 8, e + 2;
  }, i.prototype.writeUint16BE = i.prototype.writeUInt16BE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 2, 65535, 0), this[e] = t >>> 8, this[e + 1] = t & 255, e + 2;
  }, i.prototype.writeUint32LE = i.prototype.writeUInt32LE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 4, 4294967295, 0), this[e + 3] = t >>> 24, this[e + 2] = t >>> 16, this[e + 1] = t >>> 8, this[e] = t & 255, e + 4;
  }, i.prototype.writeUint32BE = i.prototype.writeUInt32BE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 4, 4294967295, 0), this[e] = t >>> 24, this[e + 1] = t >>> 16, this[e + 2] = t >>> 8, this[e + 3] = t & 255, e + 4;
  };
  function Pt(r, t, e, n, a) {
    jt(t, n, a, r, e, 7);
    let s = Number(t & BigInt(4294967295));
    r[e++] = s, s = s >> 8, r[e++] = s, s = s >> 8, r[e++] = s, s = s >> 8, r[e++] = s;
    let y = Number(t >> BigInt(32) & BigInt(4294967295));
    return r[e++] = y, y = y >> 8, r[e++] = y, y = y >> 8, r[e++] = y, y = y >> 8, r[e++] = y, e;
  }
  function Ht(r, t, e, n, a) {
    jt(t, n, a, r, e, 7);
    let s = Number(t & BigInt(4294967295));
    r[e + 7] = s, s = s >> 8, r[e + 6] = s, s = s >> 8, r[e + 5] = s, s = s >> 8, r[e + 4] = s;
    let y = Number(t >> BigInt(32) & BigInt(4294967295));
    return r[e + 3] = y, y = y >> 8, r[e + 2] = y, y = y >> 8, r[e + 1] = y, y = y >> 8, r[e] = y, e + 8;
  }
  i.prototype.writeBigUInt64LE = W(function(t, e = 0) {
    return Pt(this, t, e, BigInt(0), BigInt("0xffffffffffffffff"));
  }), i.prototype.writeBigUInt64BE = W(function(t, e = 0) {
    return Ht(this, t, e, BigInt(0), BigInt("0xffffffffffffffff"));
  }), i.prototype.writeIntLE = function(t, e, n, a) {
    if (t = +t, e = e >>> 0, !a) {
      const R = Math.pow(2, 8 * n - 1);
      H(this, t, e, n, R - 1, -R);
    }
    let s = 0, y = 1, A = 0;
    for (this[e] = t & 255; ++s < n && (y *= 256); )
      t < 0 && A === 0 && this[e + s - 1] !== 0 && (A = 1), this[e + s] = (t / y >> 0) - A & 255;
    return e + n;
  }, i.prototype.writeIntBE = function(t, e, n, a) {
    if (t = +t, e = e >>> 0, !a) {
      const R = Math.pow(2, 8 * n - 1);
      H(this, t, e, n, R - 1, -R);
    }
    let s = n - 1, y = 1, A = 0;
    for (this[e + s] = t & 255; --s >= 0 && (y *= 256); )
      t < 0 && A === 0 && this[e + s + 1] !== 0 && (A = 1), this[e + s] = (t / y >> 0) - A & 255;
    return e + n;
  }, i.prototype.writeInt8 = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 1, 127, -128), t < 0 && (t = 255 + t + 1), this[e] = t & 255, e + 1;
  }, i.prototype.writeInt16LE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 2, 32767, -32768), this[e] = t & 255, this[e + 1] = t >>> 8, e + 2;
  }, i.prototype.writeInt16BE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 2, 32767, -32768), this[e] = t >>> 8, this[e + 1] = t & 255, e + 2;
  }, i.prototype.writeInt32LE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 4, 2147483647, -2147483648), this[e] = t & 255, this[e + 1] = t >>> 8, this[e + 2] = t >>> 16, this[e + 3] = t >>> 24, e + 4;
  }, i.prototype.writeInt32BE = function(t, e, n) {
    return t = +t, e = e >>> 0, n || H(this, t, e, 4, 2147483647, -2147483648), t < 0 && (t = 4294967295 + t + 1), this[e] = t >>> 24, this[e + 1] = t >>> 16, this[e + 2] = t >>> 8, this[e + 3] = t & 255, e + 4;
  }, i.prototype.writeBigInt64LE = W(function(t, e = 0) {
    return Pt(this, t, e, -BigInt("0x8000000000000000"), BigInt("0x7fffffffffffffff"));
  }), i.prototype.writeBigInt64BE = W(function(t, e = 0) {
    return Ht(this, t, e, -BigInt("0x8000000000000000"), BigInt("0x7fffffffffffffff"));
  });
  function Ot(r, t, e, n, a, s) {
    if (e + n > r.length) throw new RangeError("Index out of range");
    if (e < 0) throw new RangeError("Index out of range");
  }
  function zt(r, t, e, n, a) {
    return t = +t, e = e >>> 0, a || Ot(r, t, e, 4), f.write(r, t, e, n, 23, 4), e + 4;
  }
  i.prototype.writeFloatLE = function(t, e, n) {
    return zt(this, t, e, !0, n);
  }, i.prototype.writeFloatBE = function(t, e, n) {
    return zt(this, t, e, !1, n);
  };
  function Wt(r, t, e, n, a) {
    return t = +t, e = e >>> 0, a || Ot(r, t, e, 8), f.write(r, t, e, n, 52, 8), e + 8;
  }
  i.prototype.writeDoubleLE = function(t, e, n) {
    return Wt(this, t, e, !0, n);
  }, i.prototype.writeDoubleBE = function(t, e, n) {
    return Wt(this, t, e, !1, n);
  }, i.prototype.copy = function(t, e, n, a) {
    if (!i.isBuffer(t)) throw new TypeError("argument should be a Buffer");
    if (n || (n = 0), !a && a !== 0 && (a = this.length), e >= t.length && (e = t.length), e || (e = 0), a > 0 && a < n && (a = n), a === n || t.length === 0 || this.length === 0) return 0;
    if (e < 0)
      throw new RangeError("targetStart out of bounds");
    if (n < 0 || n >= this.length) throw new RangeError("Index out of range");
    if (a < 0) throw new RangeError("sourceEnd out of bounds");
    a > this.length && (a = this.length), t.length - e < a - n && (a = t.length - e + n);
    const s = a - n;
    return this === t && typeof Uint8Array.prototype.copyWithin == "function" ? this.copyWithin(e, n, a) : Uint8Array.prototype.set.call(
      t,
      this.subarray(n, a),
      e
    ), s;
  }, i.prototype.fill = function(t, e, n, a) {
    if (typeof t == "string") {
      if (typeof e == "string" ? (a = e, e = 0, n = this.length) : typeof n == "string" && (a = n, n = this.length), a !== void 0 && typeof a != "string")
        throw new TypeError("encoding must be a string");
      if (typeof a == "string" && !i.isEncoding(a))
        throw new TypeError("Unknown encoding: " + a);
      if (t.length === 1) {
        const y = t.charCodeAt(0);
        (a === "utf8" && y < 128 || a === "latin1") && (t = y);
      }
    } else typeof t == "number" ? t = t & 255 : typeof t == "boolean" && (t = Number(t));
    if (e < 0 || this.length < e || this.length < n)
      throw new RangeError("Out of range index");
    if (n <= e)
      return this;
    e = e >>> 0, n = n === void 0 ? this.length : n >>> 0, t || (t = 0);
    let s;
    if (typeof t == "number")
      for (s = e; s < n; ++s)
        this[s] = t;
    else {
      const y = i.isBuffer(t) ? t : i.from(t, a), A = y.length;
      if (A === 0)
        throw new TypeError('The value "' + t + '" is invalid for argument "value"');
      for (s = 0; s < n - e; ++s)
        this[s + e] = y[s % A];
    }
    return this;
  };
  const G = {};
  function wt(r, t, e) {
    G[r] = class extends e {
      constructor() {
        super(), Object.defineProperty(this, "message", {
          value: t.apply(this, arguments),
          writable: !0,
          configurable: !0
        }), this.name = `${this.name} [${r}]`, this.stack, delete this.name;
      }
      get code() {
        return r;
      }
      set code(a) {
        Object.defineProperty(this, "code", {
          configurable: !0,
          enumerable: !0,
          value: a,
          writable: !0
        });
      }
      toString() {
        return `${this.name} [${r}]: ${this.message}`;
      }
    };
  }
  wt(
    "ERR_BUFFER_OUT_OF_BOUNDS",
    function(r) {
      return r ? `${r} is outside of buffer bounds` : "Attempt to access memory outside buffer bounds";
    },
    RangeError
  ), wt(
    "ERR_INVALID_ARG_TYPE",
    function(r, t) {
      return `The "${r}" argument must be of type number. Received type ${typeof t}`;
    },
    TypeError
  ), wt(
    "ERR_OUT_OF_RANGE",
    function(r, t, e) {
      let n = `The value of "${r}" is out of range.`, a = e;
      return Number.isInteger(e) && Math.abs(e) > 2 ** 32 ? a = qt(String(e)) : typeof e == "bigint" && (a = String(e), (e > BigInt(2) ** BigInt(32) || e < -(BigInt(2) ** BigInt(32))) && (a = qt(a)), a += "n"), n += ` It must be ${t}. Received ${a}`, n;
    },
    RangeError
  );
  function qt(r) {
    let t = "", e = r.length;
    const n = r[0] === "-" ? 1 : 0;
    for (; e >= n + 4; e -= 3)
      t = `_${r.slice(e - 3, e)}${t}`;
    return `${r.slice(0, e)}${t}`;
  }
  function ne(r, t, e) {
    K(t, "offset"), (r[t] === void 0 || r[t + e] === void 0) && Z(t, r.length - (e + 1));
  }
  function jt(r, t, e, n, a, s) {
    if (r > e || r < t) {
      const y = typeof t == "bigint" ? "n" : "";
      let A;
      throw t === 0 || t === BigInt(0) ? A = `>= 0${y} and < 2${y} ** ${(s + 1) * 8}${y}` : A = `>= -(2${y} ** ${(s + 1) * 8 - 1}${y}) and < 2 ** ${(s + 1) * 8 - 1}${y}`, new G.ERR_OUT_OF_RANGE("value", A, r);
    }
    ne(n, a, s);
  }
  function K(r, t) {
    if (typeof r != "number")
      throw new G.ERR_INVALID_ARG_TYPE(t, "number", r);
  }
  function Z(r, t, e) {
    throw Math.floor(r) !== r ? (K(r, e), new G.ERR_OUT_OF_RANGE("offset", "an integer", r)) : t < 0 ? new G.ERR_BUFFER_OUT_OF_BOUNDS() : new G.ERR_OUT_OF_RANGE(
      "offset",
      `>= 0 and <= ${t}`,
      r
    );
  }
  const ie = /[^+/0-9A-Za-z-_]/g;
  function oe(r) {
    if (r = r.split("=")[0], r = r.trim().replace(ie, ""), r.length < 2) return "";
    for (; r.length % 4 !== 0; )
      r = r + "=";
    return r;
  }
  function mt(r, t) {
    t = t || 1 / 0;
    let e;
    const n = r.length;
    let a = null;
    const s = [];
    for (let y = 0; y < n; ++y) {
      if (e = r.charCodeAt(y), e > 55295 && e < 57344) {
        if (!a) {
          if (e > 56319) {
            (t -= 3) > -1 && s.push(239, 191, 189);
            continue;
          } else if (y + 1 === n) {
            (t -= 3) > -1 && s.push(239, 191, 189);
            continue;
          }
          a = e;
          continue;
        }
        if (e < 56320) {
          (t -= 3) > -1 && s.push(239, 191, 189), a = e;
          continue;
        }
        e = (a - 55296 << 10 | e - 56320) + 65536;
      } else a && (t -= 3) > -1 && s.push(239, 191, 189);
      if (a = null, e < 128) {
        if ((t -= 1) < 0) break;
        s.push(e);
      } else if (e < 2048) {
        if ((t -= 2) < 0) break;
        s.push(
          e >> 6 | 192,
          e & 63 | 128
        );
      } else if (e < 65536) {
        if ((t -= 3) < 0) break;
        s.push(
          e >> 12 | 224,
          e >> 6 & 63 | 128,
          e & 63 | 128
        );
      } else if (e < 1114112) {
        if ((t -= 4) < 0) break;
        s.push(
          e >> 18 | 240,
          e >> 12 & 63 | 128,
          e >> 6 & 63 | 128,
          e & 63 | 128
        );
      } else
        throw new Error("Invalid code point");
    }
    return s;
  }
  function ue(r) {
    const t = [];
    for (let e = 0; e < r.length; ++e)
      t.push(r.charCodeAt(e) & 255);
    return t;
  }
  function ae(r, t) {
    let e, n, a;
    const s = [];
    for (let y = 0; y < r.length && !((t -= 2) < 0); ++y)
      e = r.charCodeAt(y), n = e >> 8, a = e % 256, s.push(a), s.push(n);
    return s;
  }
  function Gt(r) {
    return u.toByteArray(oe(r));
  }
  function lt(r, t, e, n) {
    let a;
    for (a = 0; a < n && !(a + e >= t.length || a >= r.length); ++a)
      t[a + e] = r[a];
    return a;
  }
  function O(r, t) {
    return r instanceof t || r != null && r.constructor != null && r.constructor.name != null && r.constructor.name === t.name;
  }
  function Bt(r) {
    return r !== r;
  }
  const fe = function() {
    const r = "0123456789abcdef", t = new Array(256);
    for (let e = 0; e < 16; ++e) {
      const n = e * 16;
      for (let a = 0; a < 16; ++a)
        t[n + a] = r[e] + r[a];
    }
    return t;
  }();
  function W(r) {
    return typeof BigInt > "u" ? ce : r;
  }
  function ce() {
    throw new Error("BigInt not supported");
  }
})(Se);
var Ke = {}, Ye = {}, Ve = {}, Xr = function(u, f) {
  return function(l, h) {
    var c = h * 2, d = f * 2;
    l[c] = u[d], l[c + 1] = u[d + 1];
  };
}, Ae = Xr;
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = f(Ae);
  function f(c) {
    return c && c.__esModule ? c : { default: c };
  }
  var l = function(d) {
    for (var i = d.A, B = d.C, p = 0; p < 25; p += 5) {
      for (var x = 0; x < 5; x++)
        (0, u.default)(i, p + x)(B, x);
      for (var g = 0; g < 5; g++) {
        var m = (p + g) * 2, w = (g + 1) % 5 * 2, _ = (g + 2) % 5 * 2;
        i[m] ^= ~B[w] & B[_], i[m + 1] ^= ~B[w + 1] & B[_ + 1];
      }
    }
  }, h = l;
  o.default = h;
})(Ve);
var Xe = {}, Je = {};
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = new Uint32Array([0, 1, 0, 32898, 2147483648, 32906, 2147483648, 2147516416, 0, 32907, 0, 2147483649, 2147483648, 2147516545, 2147483648, 32777, 0, 138, 0, 136, 0, 2147516425, 0, 2147483658, 0, 2147516555, 2147483648, 139, 2147483648, 32905, 2147483648, 32771, 2147483648, 32770, 2147483648, 128, 0, 32778, 2147483648, 2147483658, 2147483648, 2147516545, 2147483648, 32896, 0, 2147483649, 2147483648, 2147516424]), f = u;
  o.default = f;
})(Je);
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = f(Je);
  function f(c) {
    return c && c.__esModule ? c : { default: c };
  }
  var l = function(d) {
    var i = d.A, B = d.roundIndex, p = B * 2;
    i[0] ^= u.default[p], i[1] ^= u.default[p + 1];
  }, h = l;
  o.default = h;
})(Xe);
var Ze = {}, Qe = {};
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = [10, 7, 11, 17, 18, 3, 5, 16, 8, 21, 24, 4, 15, 23, 19, 13, 12, 2, 20, 14, 22, 9, 6, 1], f = u;
  o.default = f;
})(Qe);
var tr = {};
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = [1, 3, 6, 10, 15, 21, 28, 36, 45, 55, 2, 14, 27, 41, 56, 8, 25, 43, 62, 18, 39, 61, 20, 44], f = u;
  o.default = f;
})(tr);
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = h(Qe), f = h(tr), l = h(Ae);
  function h(i) {
    return i && i.__esModule ? i : { default: i };
  }
  var c = function(B) {
    var p = B.A, x = B.C, g = B.W;
    (0, l.default)(p, 1)(g, 0);
    for (var m = 0, w = 0, _ = 0, b = 32, v = 0; v < 24; v++) {
      var E = u.default[v], S = f.default[v];
      (0, l.default)(p, E)(x, 0), m = g[0], w = g[1], b = 32 - S, _ = S < 32 ? 0 : 1, g[_] = m << S | w >>> b, g[(_ + 1) % 2] = w << S | m >>> b, (0, l.default)(g, 0)(p, E), (0, l.default)(x, 0)(g, 0);
    }
  }, d = c;
  o.default = d;
})(Ze);
var er = {};
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = f(Ae);
  function f(c) {
    return c && c.__esModule ? c : { default: c };
  }
  var l = function(d) {
    for (var i = d.A, B = d.C, p = d.D, x = d.W, g = 0, m = 0, w = 0; w < 5; w++) {
      var _ = w * 2, b = (w + 5) * 2, v = (w + 10) * 2, E = (w + 15) * 2, S = (w + 20) * 2;
      B[_] = i[_] ^ i[b] ^ i[v] ^ i[E] ^ i[S], B[_ + 1] = i[_ + 1] ^ i[b + 1] ^ i[v + 1] ^ i[E + 1] ^ i[S + 1];
    }
    for (var I = 0; I < 5; I++) {
      (0, u.default)(B, (I + 1) % 5)(x, 0), g = x[0], m = x[1], x[0] = g << 1 | m >>> 31, x[1] = m << 1 | g >>> 31, p[I * 2] = B[(I + 4) % 5 * 2] ^ x[0], p[I * 2 + 1] = B[(I + 4) % 5 * 2 + 1] ^ x[1];
      for (var M = 0; M < 25; M += 5)
        i[(M + I) * 2] ^= p[I * 2], i[(M + I) * 2 + 1] ^= p[I * 2 + 1];
    }
  }, h = l;
  o.default = h;
})(er);
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = c(Ve), f = c(Xe), l = c(Ze), h = c(er);
  function c(B) {
    return B && B.__esModule ? B : { default: B };
  }
  var d = function() {
    var p = new Uint32Array(10), x = new Uint32Array(10), g = new Uint32Array(2);
    return function(m) {
      for (var w = 0; w < 24; w++)
        (0, h.default)({ A: m, C: p, D: x, W: g }), (0, l.default)({ A: m, C: p, W: g }), (0, u.default)({ A: m, C: p }), (0, f.default)({ A: m, roundIndex: w });
      p.fill(0), x.fill(0), g.fill(0);
    };
  }, i = d;
  o.default = i;
})(Ye);
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = void 0;
  var u = Se, f = l(Ye);
  function l(B) {
    return B && B.__esModule ? B : { default: B };
  }
  var h = function(p, x) {
    for (var g = 0; g < p.length; g += 8) {
      var m = g / 4;
      x[m] ^= p[g + 7] << 24 | p[g + 6] << 16 | p[g + 5] << 8 | p[g + 4], x[m + 1] ^= p[g + 3] << 24 | p[g + 2] << 16 | p[g + 1] << 8 | p[g];
    }
    return x;
  }, c = function(p, x) {
    for (var g = 0; g < x.length; g += 8) {
      var m = g / 4;
      x[g] = p[m + 1], x[g + 1] = p[m + 1] >>> 8, x[g + 2] = p[m + 1] >>> 16, x[g + 3] = p[m + 1] >>> 24, x[g + 4] = p[m], x[g + 5] = p[m] >>> 8, x[g + 6] = p[m] >>> 16, x[g + 7] = p[m] >>> 24;
    }
    return x;
  }, d = function(p) {
    var x = this, g = p.capacity, m = p.padding, w = (0, f.default)(), _ = 200, b = g / 8, v = _ - g / 4, E = 0, S = new Uint32Array(_ / 4), I = u.Buffer.allocUnsafe(v);
    return this.absorb = function(M) {
      for (var U = 0; U < M.length; U++)
        I[E] = M[U], E += 1, E >= v && (h(I, S), w(S), E = 0);
      return x;
    }, this.squeeze = function() {
      var M = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, U = { buffer: M.buffer || u.Buffer.allocUnsafe(b), padding: M.padding || m, queue: u.Buffer.allocUnsafe(I.length), state: new Uint32Array(S.length) };
      I.copy(U.queue);
      for (var F = 0; F < S.length; F++)
        U.state[F] = S[F];
      U.queue.fill(0, E), U.queue[E] |= U.padding, U.queue[v - 1] |= 128, h(U.queue, U.state);
      for (var D = 0; D < U.buffer.length; D += v)
        w(U.state), c(U.state, U.buffer.slice(D, D + v));
      return U.buffer;
    }, this.reset = function() {
      return I.fill(0), S.fill(0), E = 0, x;
    }, this;
  }, i = d;
  o.default = i;
})(Ke);
(function(o) {
  Object.defineProperty(o, "__esModule", { value: !0 }), o.default = o.SHAKE = o.SHA3Hash = o.SHA3 = o.Keccak = void 0;
  var u = Se, f = l(Ke);
  function l(x) {
    return x && x.__esModule ? x : { default: x };
  }
  var h = function(g) {
    var m = g.allowedSizes, w = g.defaultSize, _ = g.padding;
    return function b() {
      var v = this, E = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : w;
      if (!this || this.constructor !== b)
        return new b(E);
      if (m && !m.includes(E))
        throw new Error("Unsupported hash length");
      var S = new f.default({ capacity: E });
      return this.update = function(I) {
        var M = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "utf8";
        if (u.Buffer.isBuffer(I))
          return S.absorb(I), v;
        if (typeof I == "string")
          return v.update(u.Buffer.from(I, M));
        throw new TypeError("Not a string or buffer");
      }, this.digest = function() {
        var I = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : "binary", M = typeof I == "string" ? { format: I } : I, U = S.squeeze({ buffer: M.buffer, padding: M.padding || _ });
        return M.format && M.format !== "binary" ? U.toString(M.format) : U;
      }, this.reset = function() {
        return S.reset(), v;
      }, this;
    };
  }, c = h({ allowedSizes: [224, 256, 384, 512], defaultSize: 512, padding: 1 });
  o.Keccak = c;
  var d = h({ allowedSizes: [224, 256, 384, 512], defaultSize: 512, padding: 6 });
  o.SHA3 = d;
  var i = h({ allowedSizes: [128, 256], defaultSize: 256, padding: 31 });
  o.SHAKE = i;
  var B = c;
  o.SHA3Hash = B, d.SHA3Hash = B;
  var p = d;
  o.default = p;
})(je);
var rr = {}, N = {}, Xt = nr;
function nr(o, u) {
  if (!o)
    throw new Error(u || "Assertion failed");
}
nr.equal = function(u, f, l) {
  if (u != f)
    throw new Error(l || "Assertion failed: " + u + " != " + f);
};
var ge = { exports: {} };
typeof Object.create == "function" ? ge.exports = function(u, f) {
  f && (u.super_ = f, u.prototype = Object.create(f.prototype, {
    constructor: {
      value: u,
      enumerable: !1,
      writable: !0,
      configurable: !0
    }
  }));
} : ge.exports = function(u, f) {
  if (f) {
    u.super_ = f;
    var l = function() {
    };
    l.prototype = f.prototype, u.prototype = new l(), u.prototype.constructor = u;
  }
};
var Jr = ge.exports, Zr = Xt, Qr = Jr;
N.inherits = Qr;
function tn(o, u) {
  return (o.charCodeAt(u) & 64512) !== 55296 || u < 0 || u + 1 >= o.length ? !1 : (o.charCodeAt(u + 1) & 64512) === 56320;
}
function en(o, u) {
  if (Array.isArray(o))
    return o.slice();
  if (!o)
    return [];
  var f = [];
  if (typeof o == "string")
    if (u) {
      if (u === "hex")
        for (o = o.replace(/[^a-z0-9]+/ig, ""), o.length % 2 !== 0 && (o = "0" + o), h = 0; h < o.length; h += 2)
          f.push(parseInt(o[h] + o[h + 1], 16));
    } else for (var l = 0, h = 0; h < o.length; h++) {
      var c = o.charCodeAt(h);
      c < 128 ? f[l++] = c : c < 2048 ? (f[l++] = c >> 6 | 192, f[l++] = c & 63 | 128) : tn(o, h) ? (c = 65536 + ((c & 1023) << 10) + (o.charCodeAt(++h) & 1023), f[l++] = c >> 18 | 240, f[l++] = c >> 12 & 63 | 128, f[l++] = c >> 6 & 63 | 128, f[l++] = c & 63 | 128) : (f[l++] = c >> 12 | 224, f[l++] = c >> 6 & 63 | 128, f[l++] = c & 63 | 128);
    }
  else
    for (h = 0; h < o.length; h++)
      f[h] = o[h] | 0;
  return f;
}
N.toArray = en;
function rn(o) {
  for (var u = "", f = 0; f < o.length; f++)
    u += or(o[f].toString(16));
  return u;
}
N.toHex = rn;
function ir(o) {
  var u = o >>> 24 | o >>> 8 & 65280 | o << 8 & 16711680 | (o & 255) << 24;
  return u >>> 0;
}
N.htonl = ir;
function nn(o, u) {
  for (var f = "", l = 0; l < o.length; l++) {
    var h = o[l];
    u === "little" && (h = ir(h)), f += ur(h.toString(16));
  }
  return f;
}
N.toHex32 = nn;
function or(o) {
  return o.length === 1 ? "0" + o : o;
}
N.zero2 = or;
function ur(o) {
  return o.length === 7 ? "0" + o : o.length === 6 ? "00" + o : o.length === 5 ? "000" + o : o.length === 4 ? "0000" + o : o.length === 3 ? "00000" + o : o.length === 2 ? "000000" + o : o.length === 1 ? "0000000" + o : o;
}
N.zero8 = ur;
function on(o, u, f, l) {
  var h = f - u;
  Zr(h % 4 === 0);
  for (var c = new Array(h / 4), d = 0, i = u; d < c.length; d++, i += 4) {
    var B;
    l === "big" ? B = o[i] << 24 | o[i + 1] << 16 | o[i + 2] << 8 | o[i + 3] : B = o[i + 3] << 24 | o[i + 2] << 16 | o[i + 1] << 8 | o[i], c[d] = B >>> 0;
  }
  return c;
}
N.join32 = on;
function un(o, u) {
  for (var f = new Array(o.length * 4), l = 0, h = 0; l < o.length; l++, h += 4) {
    var c = o[l];
    u === "big" ? (f[h] = c >>> 24, f[h + 1] = c >>> 16 & 255, f[h + 2] = c >>> 8 & 255, f[h + 3] = c & 255) : (f[h + 3] = c >>> 24, f[h + 2] = c >>> 16 & 255, f[h + 1] = c >>> 8 & 255, f[h] = c & 255);
  }
  return f;
}
N.split32 = un;
function an(o, u) {
  return o >>> u | o << 32 - u;
}
N.rotr32 = an;
function fn(o, u) {
  return o << u | o >>> 32 - u;
}
N.rotl32 = fn;
function cn(o, u) {
  return o + u >>> 0;
}
N.sum32 = cn;
function hn(o, u, f) {
  return o + u + f >>> 0;
}
N.sum32_3 = hn;
function sn(o, u, f, l) {
  return o + u + f + l >>> 0;
}
N.sum32_4 = sn;
function ln(o, u, f, l, h) {
  return o + u + f + l + h >>> 0;
}
N.sum32_5 = ln;
function pn(o, u, f, l) {
  var h = o[u], c = o[u + 1], d = l + c >>> 0, i = (d < l ? 1 : 0) + f + h;
  o[u] = i >>> 0, o[u + 1] = d;
}
N.sum64 = pn;
function dn(o, u, f, l) {
  var h = u + l >>> 0, c = (h < u ? 1 : 0) + o + f;
  return c >>> 0;
}
N.sum64_hi = dn;
function yn(o, u, f, l) {
  var h = u + l;
  return h >>> 0;
}
N.sum64_lo = yn;
function xn(o, u, f, l, h, c, d, i) {
  var B = 0, p = u;
  p = p + l >>> 0, B += p < u ? 1 : 0, p = p + c >>> 0, B += p < c ? 1 : 0, p = p + i >>> 0, B += p < i ? 1 : 0;
  var x = o + f + h + d + B;
  return x >>> 0;
}
N.sum64_4_hi = xn;
function gn(o, u, f, l, h, c, d, i) {
  var B = u + l + c + i;
  return B >>> 0;
}
N.sum64_4_lo = gn;
function wn(o, u, f, l, h, c, d, i, B, p) {
  var x = 0, g = u;
  g = g + l >>> 0, x += g < u ? 1 : 0, g = g + c >>> 0, x += g < c ? 1 : 0, g = g + i >>> 0, x += g < i ? 1 : 0, g = g + p >>> 0, x += g < p ? 1 : 0;
  var m = o + f + h + d + B + x;
  return m >>> 0;
}
N.sum64_5_hi = wn;
function mn(o, u, f, l, h, c, d, i, B, p) {
  var x = u + l + c + i + p;
  return x >>> 0;
}
N.sum64_5_lo = mn;
function Bn(o, u, f) {
  var l = u << 32 - f | o >>> f;
  return l >>> 0;
}
N.rotr64_hi = Bn;
function bn(o, u, f) {
  var l = o << 32 - f | u >>> f;
  return l >>> 0;
}
N.rotr64_lo = bn;
function _n(o, u, f) {
  return o >>> f;
}
N.shr64_hi = _n;
function vn(o, u, f) {
  var l = o << 32 - f | u >>> f;
  return l >>> 0;
}
N.shr64_lo = vn;
var kt = {}, Ce = N, En = Xt;
function te() {
  this.pending = null, this.pendingTotal = 0, this.blockSize = this.constructor.blockSize, this.outSize = this.constructor.outSize, this.hmacStrength = this.constructor.hmacStrength, this.padLength = this.constructor.padLength / 8, this.endian = "big", this._delta8 = this.blockSize / 8, this._delta32 = this.blockSize / 32;
}
kt.BlockHash = te;
te.prototype.update = function(u, f) {
  if (u = Ce.toArray(u, f), this.pending ? this.pending = this.pending.concat(u) : this.pending = u, this.pendingTotal += u.length, this.pending.length >= this._delta8) {
    u = this.pending;
    var l = u.length % this._delta8;
    this.pending = u.slice(u.length - l, u.length), this.pending.length === 0 && (this.pending = null), u = Ce.join32(u, 0, u.length - l, this.endian);
    for (var h = 0; h < u.length; h += this._delta32)
      this._update(u, h, h + this._delta32);
  }
  return this;
};
te.prototype.digest = function(u) {
  return this.update(this._pad()), En(this.pending === null), this._digest(u);
};
te.prototype._pad = function() {
  var u = this.pendingTotal, f = this._delta8, l = f - (u + this.padLength) % f, h = new Array(l + this.padLength);
  h[0] = 128;
  for (var c = 1; c < l; c++)
    h[c] = 0;
  if (u <<= 3, this.endian === "big") {
    for (var d = 8; d < this.padLength; d++)
      h[c++] = 0;
    h[c++] = 0, h[c++] = 0, h[c++] = 0, h[c++] = 0, h[c++] = u >>> 24 & 255, h[c++] = u >>> 16 & 255, h[c++] = u >>> 8 & 255, h[c++] = u & 255;
  } else
    for (h[c++] = u & 255, h[c++] = u >>> 8 & 255, h[c++] = u >>> 16 & 255, h[c++] = u >>> 24 & 255, h[c++] = 0, h[c++] = 0, h[c++] = 0, h[c++] = 0, d = 8; d < this.padLength; d++)
      h[c++] = 0;
  return h;
};
var Lt = {}, st = {}, Sn = N, ot = Sn.rotr32;
function An(o, u, f, l) {
  if (o === 0)
    return ar(u, f, l);
  if (o === 1 || o === 3)
    return cr(u, f, l);
  if (o === 2)
    return fr(u, f, l);
}
st.ft_1 = An;
function ar(o, u, f) {
  return o & u ^ ~o & f;
}
st.ch32 = ar;
function fr(o, u, f) {
  return o & u ^ o & f ^ u & f;
}
st.maj32 = fr;
function cr(o, u, f) {
  return o ^ u ^ f;
}
st.p32 = cr;
function In(o) {
  return ot(o, 2) ^ ot(o, 13) ^ ot(o, 22);
}
st.s0_256 = In;
function Fn(o) {
  return ot(o, 6) ^ ot(o, 11) ^ ot(o, 25);
}
st.s1_256 = Fn;
function Un(o) {
  return ot(o, 7) ^ ot(o, 18) ^ o >>> 3;
}
st.g0_256 = Un;
function Cn(o) {
  return ot(o, 17) ^ ot(o, 19) ^ o >>> 10;
}
st.g1_256 = Cn;
var Tt = N, $n = kt, Rn = st, se = Tt.rotl32, Kt = Tt.sum32, Tn = Tt.sum32_5, Mn = Rn.ft_1, hr = $n.BlockHash, Dn = [
  1518500249,
  1859775393,
  2400959708,
  3395469782
];
function ft() {
  if (!(this instanceof ft))
    return new ft();
  hr.call(this), this.h = [
    1732584193,
    4023233417,
    2562383102,
    271733878,
    3285377520
  ], this.W = new Array(80);
}
Tt.inherits(ft, hr);
var kn = ft;
ft.blockSize = 512;
ft.outSize = 160;
ft.hmacStrength = 80;
ft.padLength = 64;
ft.prototype._update = function(u, f) {
  for (var l = this.W, h = 0; h < 16; h++)
    l[h] = u[f + h];
  for (; h < l.length; h++)
    l[h] = se(l[h - 3] ^ l[h - 8] ^ l[h - 14] ^ l[h - 16], 1);
  var c = this.h[0], d = this.h[1], i = this.h[2], B = this.h[3], p = this.h[4];
  for (h = 0; h < l.length; h++) {
    var x = ~~(h / 20), g = Tn(se(c, 5), Mn(x, d, i, B), p, l[h], Dn[x]);
    p = B, B = i, i = se(d, 30), d = c, c = g;
  }
  this.h[0] = Kt(this.h[0], c), this.h[1] = Kt(this.h[1], d), this.h[2] = Kt(this.h[2], i), this.h[3] = Kt(this.h[3], B), this.h[4] = Kt(this.h[4], p);
};
ft.prototype._digest = function(u) {
  return u === "hex" ? Tt.toHex32(this.h, "big") : Tt.split32(this.h, "big");
};
var Mt = N, Ln = kt, Nt = st, Nn = Xt, Q = Mt.sum32, Pn = Mt.sum32_4, Hn = Mt.sum32_5, On = Nt.ch32, zn = Nt.maj32, Wn = Nt.s0_256, qn = Nt.s1_256, jn = Nt.g0_256, Gn = Nt.g1_256, sr = Ln.BlockHash, Kn = [
  1116352408,
  1899447441,
  3049323471,
  3921009573,
  961987163,
  1508970993,
  2453635748,
  2870763221,
  3624381080,
  310598401,
  607225278,
  1426881987,
  1925078388,
  2162078206,
  2614888103,
  3248222580,
  3835390401,
  4022224774,
  264347078,
  604807628,
  770255983,
  1249150122,
  1555081692,
  1996064986,
  2554220882,
  2821834349,
  2952996808,
  3210313671,
  3336571891,
  3584528711,
  113926993,
  338241895,
  666307205,
  773529912,
  1294757372,
  1396182291,
  1695183700,
  1986661051,
  2177026350,
  2456956037,
  2730485921,
  2820302411,
  3259730800,
  3345764771,
  3516065817,
  3600352804,
  4094571909,
  275423344,
  430227734,
  506948616,
  659060556,
  883997877,
  958139571,
  1322822218,
  1537002063,
  1747873779,
  1955562222,
  2024104815,
  2227730452,
  2361852424,
  2428436474,
  2756734187,
  3204031479,
  3329325298
];
function ct() {
  if (!(this instanceof ct))
    return new ct();
  sr.call(this), this.h = [
    1779033703,
    3144134277,
    1013904242,
    2773480762,
    1359893119,
    2600822924,
    528734635,
    1541459225
  ], this.k = Kn, this.W = new Array(64);
}
Mt.inherits(ct, sr);
var lr = ct;
ct.blockSize = 512;
ct.outSize = 256;
ct.hmacStrength = 192;
ct.padLength = 64;
ct.prototype._update = function(u, f) {
  for (var l = this.W, h = 0; h < 16; h++)
    l[h] = u[f + h];
  for (; h < l.length; h++)
    l[h] = Pn(Gn(l[h - 2]), l[h - 7], jn(l[h - 15]), l[h - 16]);
  var c = this.h[0], d = this.h[1], i = this.h[2], B = this.h[3], p = this.h[4], x = this.h[5], g = this.h[6], m = this.h[7];
  for (Nn(this.k.length === l.length), h = 0; h < l.length; h++) {
    var w = Hn(m, qn(p), On(p, x, g), this.k[h], l[h]), _ = Q(Wn(c), zn(c, d, i));
    m = g, g = x, x = p, p = Q(B, w), B = i, i = d, d = c, c = Q(w, _);
  }
  this.h[0] = Q(this.h[0], c), this.h[1] = Q(this.h[1], d), this.h[2] = Q(this.h[2], i), this.h[3] = Q(this.h[3], B), this.h[4] = Q(this.h[4], p), this.h[5] = Q(this.h[5], x), this.h[6] = Q(this.h[6], g), this.h[7] = Q(this.h[7], m);
};
ct.prototype._digest = function(u) {
  return u === "hex" ? Mt.toHex32(this.h, "big") : Mt.split32(this.h, "big");
};
var we = N, pr = lr;
function pt() {
  if (!(this instanceof pt))
    return new pt();
  pr.call(this), this.h = [
    3238371032,
    914150663,
    812702999,
    4144912697,
    4290775857,
    1750603025,
    1694076839,
    3204075428
  ];
}
we.inherits(pt, pr);
var Yn = pt;
pt.blockSize = 512;
pt.outSize = 224;
pt.hmacStrength = 192;
pt.padLength = 64;
pt.prototype._digest = function(u) {
  return u === "hex" ? we.toHex32(this.h.slice(0, 7), "big") : we.split32(this.h.slice(0, 7), "big");
};
var j = N, Vn = kt, Xn = Xt, ut = j.rotr64_hi, at = j.rotr64_lo, dr = j.shr64_hi, yr = j.shr64_lo, gt = j.sum64, le = j.sum64_hi, pe = j.sum64_lo, Jn = j.sum64_4_hi, Zn = j.sum64_4_lo, Qn = j.sum64_5_hi, ti = j.sum64_5_lo, xr = Vn.BlockHash, ei = [
  1116352408,
  3609767458,
  1899447441,
  602891725,
  3049323471,
  3964484399,
  3921009573,
  2173295548,
  961987163,
  4081628472,
  1508970993,
  3053834265,
  2453635748,
  2937671579,
  2870763221,
  3664609560,
  3624381080,
  2734883394,
  310598401,
  1164996542,
  607225278,
  1323610764,
  1426881987,
  3590304994,
  1925078388,
  4068182383,
  2162078206,
  991336113,
  2614888103,
  633803317,
  3248222580,
  3479774868,
  3835390401,
  2666613458,
  4022224774,
  944711139,
  264347078,
  2341262773,
  604807628,
  2007800933,
  770255983,
  1495990901,
  1249150122,
  1856431235,
  1555081692,
  3175218132,
  1996064986,
  2198950837,
  2554220882,
  3999719339,
  2821834349,
  766784016,
  2952996808,
  2566594879,
  3210313671,
  3203337956,
  3336571891,
  1034457026,
  3584528711,
  2466948901,
  113926993,
  3758326383,
  338241895,
  168717936,
  666307205,
  1188179964,
  773529912,
  1546045734,
  1294757372,
  1522805485,
  1396182291,
  2643833823,
  1695183700,
  2343527390,
  1986661051,
  1014477480,
  2177026350,
  1206759142,
  2456956037,
  344077627,
  2730485921,
  1290863460,
  2820302411,
  3158454273,
  3259730800,
  3505952657,
  3345764771,
  106217008,
  3516065817,
  3606008344,
  3600352804,
  1432725776,
  4094571909,
  1467031594,
  275423344,
  851169720,
  430227734,
  3100823752,
  506948616,
  1363258195,
  659060556,
  3750685593,
  883997877,
  3785050280,
  958139571,
  3318307427,
  1322822218,
  3812723403,
  1537002063,
  2003034995,
  1747873779,
  3602036899,
  1955562222,
  1575990012,
  2024104815,
  1125592928,
  2227730452,
  2716904306,
  2361852424,
  442776044,
  2428436474,
  593698344,
  2756734187,
  3733110249,
  3204031479,
  2999351573,
  3329325298,
  3815920427,
  3391569614,
  3928383900,
  3515267271,
  566280711,
  3940187606,
  3454069534,
  4118630271,
  4000239992,
  116418474,
  1914138554,
  174292421,
  2731055270,
  289380356,
  3203993006,
  460393269,
  320620315,
  685471733,
  587496836,
  852142971,
  1086792851,
  1017036298,
  365543100,
  1126000580,
  2618297676,
  1288033470,
  3409855158,
  1501505948,
  4234509866,
  1607167915,
  987167468,
  1816402316,
  1246189591
];
function et() {
  if (!(this instanceof et))
    return new et();
  xr.call(this), this.h = [
    1779033703,
    4089235720,
    3144134277,
    2227873595,
    1013904242,
    4271175723,
    2773480762,
    1595750129,
    1359893119,
    2917565137,
    2600822924,
    725511199,
    528734635,
    4215389547,
    1541459225,
    327033209
  ], this.k = ei, this.W = new Array(160);
}
j.inherits(et, xr);
var gr = et;
et.blockSize = 1024;
et.outSize = 512;
et.hmacStrength = 192;
et.padLength = 128;
et.prototype._prepareBlock = function(u, f) {
  for (var l = this.W, h = 0; h < 32; h++)
    l[h] = u[f + h];
  for (; h < l.length; h += 2) {
    var c = li(l[h - 4], l[h - 3]), d = pi(l[h - 4], l[h - 3]), i = l[h - 14], B = l[h - 13], p = hi(l[h - 30], l[h - 29]), x = si(l[h - 30], l[h - 29]), g = l[h - 32], m = l[h - 31];
    l[h] = Jn(
      c,
      d,
      i,
      B,
      p,
      x,
      g,
      m
    ), l[h + 1] = Zn(
      c,
      d,
      i,
      B,
      p,
      x,
      g,
      m
    );
  }
};
et.prototype._update = function(u, f) {
  this._prepareBlock(u, f);
  var l = this.W, h = this.h[0], c = this.h[1], d = this.h[2], i = this.h[3], B = this.h[4], p = this.h[5], x = this.h[6], g = this.h[7], m = this.h[8], w = this.h[9], _ = this.h[10], b = this.h[11], v = this.h[12], E = this.h[13], S = this.h[14], I = this.h[15];
  Xn(this.k.length === l.length);
  for (var M = 0; M < l.length; M += 2) {
    var U = S, F = I, D = fi(m, w), P = ci(m, w), rt = ri(m, w, _, b, v), Et = ni(m, w, _, b, v, E), St = this.k[M], At = this.k[M + 1], It = l[M], yt = l[M + 1], nt = Qn(
      U,
      F,
      D,
      P,
      rt,
      Et,
      St,
      At,
      It,
      yt
    ), xt = ti(
      U,
      F,
      D,
      P,
      rt,
      Et,
      St,
      At,
      It,
      yt
    );
    U = ui(h, c), F = ai(h, c), D = ii(h, c, d, i, B), P = oi(h, c, d, i, B, p);
    var Ft = le(U, F, D, P), Ut = pe(U, F, D, P);
    S = v, I = E, v = _, E = b, _ = m, b = w, m = le(x, g, nt, xt), w = pe(g, g, nt, xt), x = B, g = p, B = d, p = i, d = h, i = c, h = le(nt, xt, Ft, Ut), c = pe(nt, xt, Ft, Ut);
  }
  gt(this.h, 0, h, c), gt(this.h, 2, d, i), gt(this.h, 4, B, p), gt(this.h, 6, x, g), gt(this.h, 8, m, w), gt(this.h, 10, _, b), gt(this.h, 12, v, E), gt(this.h, 14, S, I);
};
et.prototype._digest = function(u) {
  return u === "hex" ? j.toHex32(this.h, "big") : j.split32(this.h, "big");
};
function ri(o, u, f, l, h) {
  var c = o & f ^ ~o & h;
  return c < 0 && (c += 4294967296), c;
}
function ni(o, u, f, l, h, c) {
  var d = u & l ^ ~u & c;
  return d < 0 && (d += 4294967296), d;
}
function ii(o, u, f, l, h) {
  var c = o & f ^ o & h ^ f & h;
  return c < 0 && (c += 4294967296), c;
}
function oi(o, u, f, l, h, c) {
  var d = u & l ^ u & c ^ l & c;
  return d < 0 && (d += 4294967296), d;
}
function ui(o, u) {
  var f = ut(o, u, 28), l = ut(u, o, 2), h = ut(u, o, 7), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function ai(o, u) {
  var f = at(o, u, 28), l = at(u, o, 2), h = at(u, o, 7), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function fi(o, u) {
  var f = ut(o, u, 14), l = ut(o, u, 18), h = ut(u, o, 9), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function ci(o, u) {
  var f = at(o, u, 14), l = at(o, u, 18), h = at(u, o, 9), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function hi(o, u) {
  var f = ut(o, u, 1), l = ut(o, u, 8), h = dr(o, u, 7), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function si(o, u) {
  var f = at(o, u, 1), l = at(o, u, 8), h = yr(o, u, 7), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function li(o, u) {
  var f = ut(o, u, 19), l = ut(u, o, 29), h = dr(o, u, 6), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
function pi(o, u) {
  var f = at(o, u, 19), l = at(u, o, 29), h = yr(o, u, 6), c = f ^ l ^ h;
  return c < 0 && (c += 4294967296), c;
}
var me = N, wr = gr;
function dt() {
  if (!(this instanceof dt))
    return new dt();
  wr.call(this), this.h = [
    3418070365,
    3238371032,
    1654270250,
    914150663,
    2438529370,
    812702999,
    355462360,
    4144912697,
    1731405415,
    4290775857,
    2394180231,
    1750603025,
    3675008525,
    1694076839,
    1203062813,
    3204075428
  ];
}
me.inherits(dt, wr);
var di = dt;
dt.blockSize = 1024;
dt.outSize = 384;
dt.hmacStrength = 192;
dt.padLength = 128;
dt.prototype._digest = function(u) {
  return u === "hex" ? me.toHex32(this.h.slice(0, 12), "big") : me.split32(this.h.slice(0, 12), "big");
};
Lt.sha1 = kn;
Lt.sha224 = Yn;
Lt.sha256 = lr;
Lt.sha384 = di;
Lt.sha512 = gr;
var mr = {}, _t = N, yi = kt, Jt = _t.rotl32, $e = _t.sum32, Yt = _t.sum32_3, Re = _t.sum32_4, Br = yi.BlockHash;
function ht() {
  if (!(this instanceof ht))
    return new ht();
  Br.call(this), this.h = [1732584193, 4023233417, 2562383102, 271733878, 3285377520], this.endian = "little";
}
_t.inherits(ht, Br);
mr.ripemd160 = ht;
ht.blockSize = 512;
ht.outSize = 160;
ht.hmacStrength = 192;
ht.padLength = 64;
ht.prototype._update = function(u, f) {
  for (var l = this.h[0], h = this.h[1], c = this.h[2], d = this.h[3], i = this.h[4], B = l, p = h, x = c, g = d, m = i, w = 0; w < 80; w++) {
    var _ = $e(
      Jt(
        Re(l, Te(w, h, c, d), u[wi[w] + f], xi(w)),
        Bi[w]
      ),
      i
    );
    l = i, i = d, d = Jt(c, 10), c = h, h = _, _ = $e(
      Jt(
        Re(B, Te(79 - w, p, x, g), u[mi[w] + f], gi(w)),
        bi[w]
      ),
      m
    ), B = m, m = g, g = Jt(x, 10), x = p, p = _;
  }
  _ = Yt(this.h[1], c, g), this.h[1] = Yt(this.h[2], d, m), this.h[2] = Yt(this.h[3], i, B), this.h[3] = Yt(this.h[4], l, p), this.h[4] = Yt(this.h[0], h, x), this.h[0] = _;
};
ht.prototype._digest = function(u) {
  return u === "hex" ? _t.toHex32(this.h, "little") : _t.split32(this.h, "little");
};
function Te(o, u, f, l) {
  return o <= 15 ? u ^ f ^ l : o <= 31 ? u & f | ~u & l : o <= 47 ? (u | ~f) ^ l : o <= 63 ? u & l | f & ~l : u ^ (f | ~l);
}
function xi(o) {
  return o <= 15 ? 0 : o <= 31 ? 1518500249 : o <= 47 ? 1859775393 : o <= 63 ? 2400959708 : 2840853838;
}
function gi(o) {
  return o <= 15 ? 1352829926 : o <= 31 ? 1548603684 : o <= 47 ? 1836072691 : o <= 63 ? 2053994217 : 0;
}
var wi = [
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
  13,
  14,
  15,
  7,
  4,
  13,
  1,
  10,
  6,
  15,
  3,
  12,
  0,
  9,
  5,
  2,
  14,
  11,
  8,
  3,
  10,
  14,
  4,
  9,
  15,
  8,
  1,
  2,
  7,
  0,
  6,
  13,
  11,
  5,
  12,
  1,
  9,
  11,
  10,
  0,
  8,
  12,
  4,
  13,
  3,
  7,
  15,
  14,
  5,
  6,
  2,
  4,
  0,
  5,
  9,
  7,
  12,
  2,
  10,
  14,
  1,
  3,
  8,
  11,
  6,
  15,
  13
], mi = [
  5,
  14,
  7,
  0,
  9,
  2,
  11,
  4,
  13,
  6,
  15,
  8,
  1,
  10,
  3,
  12,
  6,
  11,
  3,
  7,
  0,
  13,
  5,
  10,
  14,
  15,
  8,
  12,
  4,
  9,
  1,
  2,
  15,
  5,
  1,
  3,
  7,
  14,
  6,
  9,
  11,
  8,
  12,
  2,
  10,
  0,
  4,
  13,
  8,
  6,
  4,
  1,
  3,
  11,
  15,
  0,
  5,
  12,
  2,
  13,
  9,
  7,
  10,
  14,
  12,
  15,
  10,
  4,
  1,
  5,
  8,
  7,
  6,
  2,
  13,
  14,
  0,
  3,
  9,
  11
], Bi = [
  11,
  14,
  15,
  12,
  5,
  8,
  7,
  9,
  11,
  13,
  14,
  15,
  6,
  7,
  9,
  8,
  7,
  6,
  8,
  13,
  11,
  9,
  7,
  15,
  7,
  12,
  15,
  9,
  11,
  7,
  13,
  12,
  11,
  13,
  6,
  7,
  14,
  9,
  13,
  15,
  14,
  8,
  13,
  6,
  5,
  12,
  7,
  5,
  11,
  12,
  14,
  15,
  14,
  15,
  9,
  8,
  9,
  14,
  5,
  6,
  8,
  6,
  5,
  12,
  9,
  15,
  5,
  11,
  6,
  8,
  13,
  12,
  5,
  12,
  13,
  14,
  11,
  8,
  5,
  6
], bi = [
  8,
  9,
  9,
  11,
  13,
  15,
  15,
  5,
  7,
  7,
  8,
  11,
  14,
  14,
  12,
  6,
  9,
  13,
  15,
  7,
  12,
  8,
  9,
  11,
  7,
  7,
  12,
  7,
  6,
  15,
  13,
  11,
  9,
  7,
  15,
  11,
  8,
  6,
  6,
  14,
  12,
  13,
  5,
  14,
  13,
  13,
  7,
  5,
  15,
  5,
  8,
  11,
  14,
  14,
  6,
  14,
  6,
  9,
  12,
  9,
  12,
  5,
  15,
  8,
  8,
  5,
  12,
  9,
  12,
  5,
  14,
  6,
  8,
  13,
  6,
  5,
  15,
  13,
  11,
  11
], _i = N, vi = Xt;
function Dt(o, u, f) {
  if (!(this instanceof Dt))
    return new Dt(o, u, f);
  this.Hash = o, this.blockSize = o.blockSize / 8, this.outSize = o.outSize / 8, this.inner = null, this.outer = null, this._init(_i.toArray(u, f));
}
var Ei = Dt;
Dt.prototype._init = function(u) {
  u.length > this.blockSize && (u = new this.Hash().update(u).digest()), vi(u.length <= this.blockSize);
  for (var f = u.length; f < this.blockSize; f++)
    u.push(0);
  for (f = 0; f < u.length; f++)
    u[f] ^= 54;
  for (this.inner = new this.Hash().update(u), f = 0; f < u.length; f++)
    u[f] ^= 106;
  this.outer = new this.Hash().update(u);
};
Dt.prototype.update = function(u, f) {
  return this.inner.update(u, f), this;
};
Dt.prototype.digest = function(u) {
  return this.outer.update(this.inner.digest()), this.outer.digest(u);
};
(function(o) {
  var u = o;
  u.utils = N, u.common = kt, u.sha = Lt, u.ripemd = mr, u.hmac = Ei, u.sha1 = u.sha.sha1, u.sha256 = u.sha.sha256, u.sha224 = u.sha.sha224, u.sha384 = u.sha.sha384, u.sha512 = u.sha.sha512, u.ripemd160 = u.ripemd.ripemd160;
})(rr);
var de = {};
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
var Me;
function Si() {
  return Me || (Me = 1, function(o) {
    const u = Vt, f = Qt, l = typeof Symbol == "function" && typeof Symbol.for == "function" ? Symbol.for("nodejs.util.inspect.custom") : null;
    o.Buffer = i, o.SlowBuffer = S, o.INSPECT_MAX_BYTES = 50;
    const h = 2147483647;
    o.kMaxLength = h, i.TYPED_ARRAY_SUPPORT = c(), !i.TYPED_ARRAY_SUPPORT && typeof console < "u" && typeof console.error == "function" && console.error(
      "This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support."
    );
    function c() {
      try {
        const r = new Uint8Array(1), t = { foo: function() {
          return 42;
        } };
        return Object.setPrototypeOf(t, Uint8Array.prototype), Object.setPrototypeOf(r, t), r.foo() === 42;
      } catch {
        return !1;
      }
    }
    Object.defineProperty(i.prototype, "parent", {
      enumerable: !0,
      get: function() {
        if (i.isBuffer(this))
          return this.buffer;
      }
    }), Object.defineProperty(i.prototype, "offset", {
      enumerable: !0,
      get: function() {
        if (i.isBuffer(this))
          return this.byteOffset;
      }
    });
    function d(r) {
      if (r > h)
        throw new RangeError('The value "' + r + '" is invalid for option "size"');
      const t = new Uint8Array(r);
      return Object.setPrototypeOf(t, i.prototype), t;
    }
    function i(r, t, e) {
      if (typeof r == "number") {
        if (typeof t == "string")
          throw new TypeError(
            'The "string" argument must be of type string. Received type number'
          );
        return g(r);
      }
      return B(r, t, e);
    }
    i.poolSize = 8192;
    function B(r, t, e) {
      if (typeof r == "string")
        return m(r, t);
      if (ArrayBuffer.isView(r))
        return _(r);
      if (r == null)
        throw new TypeError(
          "The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type " + typeof r
        );
      if (O(r, ArrayBuffer) || r && O(r.buffer, ArrayBuffer) || typeof SharedArrayBuffer < "u" && (O(r, SharedArrayBuffer) || r && O(r.buffer, SharedArrayBuffer)))
        return b(r, t, e);
      if (typeof r == "number")
        throw new TypeError(
          'The "value" argument must not be of type number. Received type number'
        );
      const n = r.valueOf && r.valueOf();
      if (n != null && n !== r)
        return i.from(n, t, e);
      const a = v(r);
      if (a) return a;
      if (typeof Symbol < "u" && Symbol.toPrimitive != null && typeof r[Symbol.toPrimitive] == "function")
        return i.from(r[Symbol.toPrimitive]("string"), t, e);
      throw new TypeError(
        "The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type " + typeof r
      );
    }
    i.from = function(r, t, e) {
      return B(r, t, e);
    }, Object.setPrototypeOf(i.prototype, Uint8Array.prototype), Object.setPrototypeOf(i, Uint8Array);
    function p(r) {
      if (typeof r != "number")
        throw new TypeError('"size" argument must be of type number');
      if (r < 0)
        throw new RangeError('The value "' + r + '" is invalid for option "size"');
    }
    function x(r, t, e) {
      return p(r), r <= 0 ? d(r) : t !== void 0 ? typeof e == "string" ? d(r).fill(t, e) : d(r).fill(t) : d(r);
    }
    i.alloc = function(r, t, e) {
      return x(r, t, e);
    };
    function g(r) {
      return p(r), d(r < 0 ? 0 : E(r) | 0);
    }
    i.allocUnsafe = function(r) {
      return g(r);
    }, i.allocUnsafeSlow = function(r) {
      return g(r);
    };
    function m(r, t) {
      if ((typeof t != "string" || t === "") && (t = "utf8"), !i.isEncoding(t))
        throw new TypeError("Unknown encoding: " + t);
      const e = I(r, t) | 0;
      let n = d(e);
      const a = n.write(r, t);
      return a !== e && (n = n.slice(0, a)), n;
    }
    function w(r) {
      const t = r.length < 0 ? 0 : E(r.length) | 0, e = d(t);
      for (let n = 0; n < t; n += 1)
        e[n] = r[n] & 255;
      return e;
    }
    function _(r) {
      if (O(r, Uint8Array)) {
        const t = new Uint8Array(r);
        return b(t.buffer, t.byteOffset, t.byteLength);
      }
      return w(r);
    }
    function b(r, t, e) {
      if (t < 0 || r.byteLength < t)
        throw new RangeError('"offset" is outside of buffer bounds');
      if (r.byteLength < t + (e || 0))
        throw new RangeError('"length" is outside of buffer bounds');
      let n;
      return t === void 0 && e === void 0 ? n = new Uint8Array(r) : e === void 0 ? n = new Uint8Array(r, t) : n = new Uint8Array(r, t, e), Object.setPrototypeOf(n, i.prototype), n;
    }
    function v(r) {
      if (i.isBuffer(r)) {
        const t = E(r.length) | 0, e = d(t);
        return e.length === 0 || r.copy(e, 0, 0, t), e;
      }
      if (r.length !== void 0)
        return typeof r.length != "number" || Bt(r.length) ? d(0) : w(r);
      if (r.type === "Buffer" && Array.isArray(r.data))
        return w(r.data);
    }
    function E(r) {
      if (r >= h)
        throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x" + h.toString(16) + " bytes");
      return r | 0;
    }
    function S(r) {
      return +r != r && (r = 0), i.alloc(+r);
    }
    i.isBuffer = function(t) {
      return t != null && t._isBuffer === !0 && t !== i.prototype;
    }, i.compare = function(t, e) {
      if (O(t, Uint8Array) && (t = i.from(t, t.offset, t.byteLength)), O(e, Uint8Array) && (e = i.from(e, e.offset, e.byteLength)), !i.isBuffer(t) || !i.isBuffer(e))
        throw new TypeError(
          'The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array'
        );
      if (t === e) return 0;
      let n = t.length, a = e.length;
      for (let s = 0, y = Math.min(n, a); s < y; ++s)
        if (t[s] !== e[s]) {
          n = t[s], a = e[s];
          break;
        }
      return n < a ? -1 : a < n ? 1 : 0;
    }, i.isEncoding = function(t) {
      switch (String(t).toLowerCase()) {
        case "hex":
        case "utf8":
        case "utf-8":
        case "ascii":
        case "latin1":
        case "binary":
        case "base64":
        case "ucs2":
        case "ucs-2":
        case "utf16le":
        case "utf-16le":
          return !0;
        default:
          return !1;
      }
    }, i.concat = function(t, e) {
      if (!Array.isArray(t))
        throw new TypeError('"list" argument must be an Array of Buffers');
      if (t.length === 0)
        return i.alloc(0);
      let n;
      if (e === void 0)
        for (e = 0, n = 0; n < t.length; ++n)
          e += t[n].length;
      const a = i.allocUnsafe(e);
      let s = 0;
      for (n = 0; n < t.length; ++n) {
        let y = t[n];
        if (O(y, Uint8Array))
          s + y.length > a.length ? (i.isBuffer(y) || (y = i.from(y)), y.copy(a, s)) : Uint8Array.prototype.set.call(
            a,
            y,
            s
          );
        else if (i.isBuffer(y))
          y.copy(a, s);
        else
          throw new TypeError('"list" argument must be an Array of Buffers');
        s += y.length;
      }
      return a;
    };
    function I(r, t) {
      if (i.isBuffer(r))
        return r.length;
      if (ArrayBuffer.isView(r) || O(r, ArrayBuffer))
        return r.byteLength;
      if (typeof r != "string")
        throw new TypeError(
          'The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type ' + typeof r
        );
      const e = r.length, n = arguments.length > 2 && arguments[2] === !0;
      if (!n && e === 0) return 0;
      let a = !1;
      for (; ; )
        switch (t) {
          case "ascii":
          case "latin1":
          case "binary":
            return e;
          case "utf8":
          case "utf-8":
            return mt(r).length;
          case "ucs2":
          case "ucs-2":
          case "utf16le":
          case "utf-16le":
            return e * 2;
          case "hex":
            return e >>> 1;
          case "base64":
            return Gt(r).length;
          default:
            if (a)
              return n ? -1 : mt(r).length;
            t = ("" + t).toLowerCase(), a = !0;
        }
    }
    i.byteLength = I;
    function M(r, t, e) {
      let n = !1;
      if ((t === void 0 || t < 0) && (t = 0), t > this.length || ((e === void 0 || e > this.length) && (e = this.length), e <= 0) || (e >>>= 0, t >>>= 0, e <= t))
        return "";
      for (r || (r = "utf8"); ; )
        switch (r) {
          case "hex":
            return ee(this, t, e);
          case "utf8":
          case "utf-8":
            return yt(this, t, e);
          case "ascii":
            return Ft(this, t, e);
          case "latin1":
          case "binary":
            return Ut(this, t, e);
          case "base64":
            return It(this, t, e);
          case "ucs2":
          case "ucs-2":
          case "utf16le":
          case "utf-16le":
            return re(this, t, e);
          default:
            if (n) throw new TypeError("Unknown encoding: " + r);
            r = (r + "").toLowerCase(), n = !0;
        }
    }
    i.prototype._isBuffer = !0;
    function U(r, t, e) {
      const n = r[t];
      r[t] = r[e], r[e] = n;
    }
    i.prototype.swap16 = function() {
      const t = this.length;
      if (t % 2 !== 0)
        throw new RangeError("Buffer size must be a multiple of 16-bits");
      for (let e = 0; e < t; e += 2)
        U(this, e, e + 1);
      return this;
    }, i.prototype.swap32 = function() {
      const t = this.length;
      if (t % 4 !== 0)
        throw new RangeError("Buffer size must be a multiple of 32-bits");
      for (let e = 0; e < t; e += 4)
        U(this, e, e + 3), U(this, e + 1, e + 2);
      return this;
    }, i.prototype.swap64 = function() {
      const t = this.length;
      if (t % 8 !== 0)
        throw new RangeError("Buffer size must be a multiple of 64-bits");
      for (let e = 0; e < t; e += 8)
        U(this, e, e + 7), U(this, e + 1, e + 6), U(this, e + 2, e + 5), U(this, e + 3, e + 4);
      return this;
    }, i.prototype.toString = function() {
      const t = this.length;
      return t === 0 ? "" : arguments.length === 0 ? yt(this, 0, t) : M.apply(this, arguments);
    }, i.prototype.toLocaleString = i.prototype.toString, i.prototype.equals = function(t) {
      if (!i.isBuffer(t)) throw new TypeError("Argument must be a Buffer");
      return this === t ? !0 : i.compare(this, t) === 0;
    }, i.prototype.inspect = function() {
      let t = "";
      const e = o.INSPECT_MAX_BYTES;
      return t = this.toString("hex", 0, e).replace(/(.{2})/g, "$1 ").trim(), this.length > e && (t += " ... "), "<Buffer " + t + ">";
    }, l && (i.prototype[l] = i.prototype.inspect), i.prototype.compare = function(t, e, n, a, s) {
      if (O(t, Uint8Array) && (t = i.from(t, t.offset, t.byteLength)), !i.isBuffer(t))
        throw new TypeError(
          'The "target" argument must be one of type Buffer or Uint8Array. Received type ' + typeof t
        );
      if (e === void 0 && (e = 0), n === void 0 && (n = t ? t.length : 0), a === void 0 && (a = 0), s === void 0 && (s = this.length), e < 0 || n > t.length || a < 0 || s > this.length)
        throw new RangeError("out of range index");
      if (a >= s && e >= n)
        return 0;
      if (a >= s)
        return -1;
      if (e >= n)
        return 1;
      if (e >>>= 0, n >>>= 0, a >>>= 0, s >>>= 0, this === t) return 0;
      let y = s - a, A = n - e;
      const R = Math.min(y, A), $ = this.slice(a, s), T = t.slice(e, n);
      for (let C = 0; C < R; ++C)
        if ($[C] !== T[C]) {
          y = $[C], A = T[C];
          break;
        }
      return y < A ? -1 : A < y ? 1 : 0;
    };
    function F(r, t, e, n, a) {
      if (r.length === 0) return -1;
      if (typeof e == "string" ? (n = e, e = 0) : e > 2147483647 ? e = 2147483647 : e < -2147483648 && (e = -2147483648), e = +e, Bt(e) && (e = a ? 0 : r.length - 1), e < 0 && (e = r.length + e), e >= r.length) {
        if (a) return -1;
        e = r.length - 1;
      } else if (e < 0)
        if (a) e = 0;
        else return -1;
      if (typeof t == "string" && (t = i.from(t, n)), i.isBuffer(t))
        return t.length === 0 ? -1 : D(r, t, e, n, a);
      if (typeof t == "number")
        return t = t & 255, typeof Uint8Array.prototype.indexOf == "function" ? a ? Uint8Array.prototype.indexOf.call(r, t, e) : Uint8Array.prototype.lastIndexOf.call(r, t, e) : D(r, [t], e, n, a);
      throw new TypeError("val must be string, number or Buffer");
    }
    function D(r, t, e, n, a) {
      let s = 1, y = r.length, A = t.length;
      if (n !== void 0 && (n = String(n).toLowerCase(), n === "ucs2" || n === "ucs-2" || n === "utf16le" || n === "utf-16le")) {
        if (r.length < 2 || t.length < 2)
          return -1;
        s = 2, y /= 2, A /= 2, e /= 2;
      }
      function R(T, C) {
        return s === 1 ? T[C] : T.readUInt16BE(C * s);
      }
      let $;
      if (a) {
        let T = -1;
        for ($ = e; $ < y; $++)
          if (R(r, $) === R(t, T === -1 ? 0 : $ - T)) {
            if (T === -1 && (T = $), $ - T + 1 === A) return T * s;
          } else
            T !== -1 && ($ -= $ - T), T = -1;
      } else
        for (e + A > y && (e = y - A), $ = e; $ >= 0; $--) {
          let T = !0;
          for (let C = 0; C < A; C++)
            if (R(r, $ + C) !== R(t, C)) {
              T = !1;
              break;
            }
          if (T) return $;
        }
      return -1;
    }
    i.prototype.includes = function(t, e, n) {
      return this.indexOf(t, e, n) !== -1;
    }, i.prototype.indexOf = function(t, e, n) {
      return F(this, t, e, n, !0);
    }, i.prototype.lastIndexOf = function(t, e, n) {
      return F(this, t, e, n, !1);
    };
    function P(r, t, e, n) {
      e = Number(e) || 0;
      const a = r.length - e;
      n ? (n = Number(n), n > a && (n = a)) : n = a;
      const s = t.length;
      n > s / 2 && (n = s / 2);
      let y;
      for (y = 0; y < n; ++y) {
        const A = parseInt(t.substr(y * 2, 2), 16);
        if (Bt(A)) return y;
        r[e + y] = A;
      }
      return y;
    }
    function rt(r, t, e, n) {
      return lt(mt(t, r.length - e), r, e, n);
    }
    function Et(r, t, e, n) {
      return lt(ue(t), r, e, n);
    }
    function St(r, t, e, n) {
      return lt(Gt(t), r, e, n);
    }
    function At(r, t, e, n) {
      return lt(ae(t, r.length - e), r, e, n);
    }
    i.prototype.write = function(t, e, n, a) {
      if (e === void 0)
        a = "utf8", n = this.length, e = 0;
      else if (n === void 0 && typeof e == "string")
        a = e, n = this.length, e = 0;
      else if (isFinite(e))
        e = e >>> 0, isFinite(n) ? (n = n >>> 0, a === void 0 && (a = "utf8")) : (a = n, n = void 0);
      else
        throw new Error(
          "Buffer.write(string, encoding, offset[, length]) is no longer supported"
        );
      const s = this.length - e;
      if ((n === void 0 || n > s) && (n = s), t.length > 0 && (n < 0 || e < 0) || e > this.length)
        throw new RangeError("Attempt to write outside buffer bounds");
      a || (a = "utf8");
      let y = !1;
      for (; ; )
        switch (a) {
          case "hex":
            return P(this, t, e, n);
          case "utf8":
          case "utf-8":
            return rt(this, t, e, n);
          case "ascii":
          case "latin1":
          case "binary":
            return Et(this, t, e, n);
          case "base64":
            return St(this, t, e, n);
          case "ucs2":
          case "ucs-2":
          case "utf16le":
          case "utf-16le":
            return At(this, t, e, n);
          default:
            if (y) throw new TypeError("Unknown encoding: " + a);
            a = ("" + a).toLowerCase(), y = !0;
        }
    }, i.prototype.toJSON = function() {
      return {
        type: "Buffer",
        data: Array.prototype.slice.call(this._arr || this, 0)
      };
    };
    function It(r, t, e) {
      return t === 0 && e === r.length ? u.fromByteArray(r) : u.fromByteArray(r.slice(t, e));
    }
    function yt(r, t, e) {
      e = Math.min(r.length, e);
      const n = [];
      let a = t;
      for (; a < e; ) {
        const s = r[a];
        let y = null, A = s > 239 ? 4 : s > 223 ? 3 : s > 191 ? 2 : 1;
        if (a + A <= e) {
          let R, $, T, C;
          switch (A) {
            case 1:
              s < 128 && (y = s);
              break;
            case 2:
              R = r[a + 1], (R & 192) === 128 && (C = (s & 31) << 6 | R & 63, C > 127 && (y = C));
              break;
            case 3:
              R = r[a + 1], $ = r[a + 2], (R & 192) === 128 && ($ & 192) === 128 && (C = (s & 15) << 12 | (R & 63) << 6 | $ & 63, C > 2047 && (C < 55296 || C > 57343) && (y = C));
              break;
            case 4:
              R = r[a + 1], $ = r[a + 2], T = r[a + 3], (R & 192) === 128 && ($ & 192) === 128 && (T & 192) === 128 && (C = (s & 15) << 18 | (R & 63) << 12 | ($ & 63) << 6 | T & 63, C > 65535 && C < 1114112 && (y = C));
          }
        }
        y === null ? (y = 65533, A = 1) : y > 65535 && (y -= 65536, n.push(y >>> 10 & 1023 | 55296), y = 56320 | y & 1023), n.push(y), a += A;
      }
      return xt(n);
    }
    const nt = 4096;
    function xt(r) {
      const t = r.length;
      if (t <= nt)
        return String.fromCharCode.apply(String, r);
      let e = "", n = 0;
      for (; n < t; )
        e += String.fromCharCode.apply(
          String,
          r.slice(n, n += nt)
        );
      return e;
    }
    function Ft(r, t, e) {
      let n = "";
      e = Math.min(r.length, e);
      for (let a = t; a < e; ++a)
        n += String.fromCharCode(r[a] & 127);
      return n;
    }
    function Ut(r, t, e) {
      let n = "";
      e = Math.min(r.length, e);
      for (let a = t; a < e; ++a)
        n += String.fromCharCode(r[a]);
      return n;
    }
    function ee(r, t, e) {
      const n = r.length;
      (!t || t < 0) && (t = 0), (!e || e < 0 || e > n) && (e = n);
      let a = "";
      for (let s = t; s < e; ++s)
        a += fe[r[s]];
      return a;
    }
    function re(r, t, e) {
      const n = r.slice(t, e);
      let a = "";
      for (let s = 0; s < n.length - 1; s += 2)
        a += String.fromCharCode(n[s] + n[s + 1] * 256);
      return a;
    }
    i.prototype.slice = function(t, e) {
      const n = this.length;
      t = ~~t, e = e === void 0 ? n : ~~e, t < 0 ? (t += n, t < 0 && (t = 0)) : t > n && (t = n), e < 0 ? (e += n, e < 0 && (e = 0)) : e > n && (e = n), e < t && (e = t);
      const a = this.subarray(t, e);
      return Object.setPrototypeOf(a, i.prototype), a;
    };
    function k(r, t, e) {
      if (r % 1 !== 0 || r < 0) throw new RangeError("offset is not uint");
      if (r + t > e) throw new RangeError("Trying to access beyond buffer length");
    }
    i.prototype.readUintLE = i.prototype.readUIntLE = function(t, e, n) {
      t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
      let a = this[t], s = 1, y = 0;
      for (; ++y < e && (s *= 256); )
        a += this[t + y] * s;
      return a;
    }, i.prototype.readUintBE = i.prototype.readUIntBE = function(t, e, n) {
      t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
      let a = this[t + --e], s = 1;
      for (; e > 0 && (s *= 256); )
        a += this[t + --e] * s;
      return a;
    }, i.prototype.readUint8 = i.prototype.readUInt8 = function(t, e) {
      return t = t >>> 0, e || k(t, 1, this.length), this[t];
    }, i.prototype.readUint16LE = i.prototype.readUInt16LE = function(t, e) {
      return t = t >>> 0, e || k(t, 2, this.length), this[t] | this[t + 1] << 8;
    }, i.prototype.readUint16BE = i.prototype.readUInt16BE = function(t, e) {
      return t = t >>> 0, e || k(t, 2, this.length), this[t] << 8 | this[t + 1];
    }, i.prototype.readUint32LE = i.prototype.readUInt32LE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), (this[t] | this[t + 1] << 8 | this[t + 2] << 16) + this[t + 3] * 16777216;
    }, i.prototype.readUint32BE = i.prototype.readUInt32BE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), this[t] * 16777216 + (this[t + 1] << 16 | this[t + 2] << 8 | this[t + 3]);
    }, i.prototype.readBigUInt64LE = W(function(t) {
      t = t >>> 0, K(t, "offset");
      const e = this[t], n = this[t + 7];
      (e === void 0 || n === void 0) && Z(t, this.length - 8);
      const a = e + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + this[++t] * 2 ** 24, s = this[++t] + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + n * 2 ** 24;
      return BigInt(a) + (BigInt(s) << BigInt(32));
    }), i.prototype.readBigUInt64BE = W(function(t) {
      t = t >>> 0, K(t, "offset");
      const e = this[t], n = this[t + 7];
      (e === void 0 || n === void 0) && Z(t, this.length - 8);
      const a = e * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + this[++t], s = this[++t] * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + n;
      return (BigInt(a) << BigInt(32)) + BigInt(s);
    }), i.prototype.readIntLE = function(t, e, n) {
      t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
      let a = this[t], s = 1, y = 0;
      for (; ++y < e && (s *= 256); )
        a += this[t + y] * s;
      return s *= 128, a >= s && (a -= Math.pow(2, 8 * e)), a;
    }, i.prototype.readIntBE = function(t, e, n) {
      t = t >>> 0, e = e >>> 0, n || k(t, e, this.length);
      let a = e, s = 1, y = this[t + --a];
      for (; a > 0 && (s *= 256); )
        y += this[t + --a] * s;
      return s *= 128, y >= s && (y -= Math.pow(2, 8 * e)), y;
    }, i.prototype.readInt8 = function(t, e) {
      return t = t >>> 0, e || k(t, 1, this.length), this[t] & 128 ? (255 - this[t] + 1) * -1 : this[t];
    }, i.prototype.readInt16LE = function(t, e) {
      t = t >>> 0, e || k(t, 2, this.length);
      const n = this[t] | this[t + 1] << 8;
      return n & 32768 ? n | 4294901760 : n;
    }, i.prototype.readInt16BE = function(t, e) {
      t = t >>> 0, e || k(t, 2, this.length);
      const n = this[t + 1] | this[t] << 8;
      return n & 32768 ? n | 4294901760 : n;
    }, i.prototype.readInt32LE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), this[t] | this[t + 1] << 8 | this[t + 2] << 16 | this[t + 3] << 24;
    }, i.prototype.readInt32BE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), this[t] << 24 | this[t + 1] << 16 | this[t + 2] << 8 | this[t + 3];
    }, i.prototype.readBigInt64LE = W(function(t) {
      t = t >>> 0, K(t, "offset");
      const e = this[t], n = this[t + 7];
      (e === void 0 || n === void 0) && Z(t, this.length - 8);
      const a = this[t + 4] + this[t + 5] * 2 ** 8 + this[t + 6] * 2 ** 16 + (n << 24);
      return (BigInt(a) << BigInt(32)) + BigInt(e + this[++t] * 2 ** 8 + this[++t] * 2 ** 16 + this[++t] * 2 ** 24);
    }), i.prototype.readBigInt64BE = W(function(t) {
      t = t >>> 0, K(t, "offset");
      const e = this[t], n = this[t + 7];
      (e === void 0 || n === void 0) && Z(t, this.length - 8);
      const a = (e << 24) + // Overflow
      this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + this[++t];
      return (BigInt(a) << BigInt(32)) + BigInt(this[++t] * 2 ** 24 + this[++t] * 2 ** 16 + this[++t] * 2 ** 8 + n);
    }), i.prototype.readFloatLE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), f.read(this, t, !0, 23, 4);
    }, i.prototype.readFloatBE = function(t, e) {
      return t = t >>> 0, e || k(t, 4, this.length), f.read(this, t, !1, 23, 4);
    }, i.prototype.readDoubleLE = function(t, e) {
      return t = t >>> 0, e || k(t, 8, this.length), f.read(this, t, !0, 52, 8);
    }, i.prototype.readDoubleBE = function(t, e) {
      return t = t >>> 0, e || k(t, 8, this.length), f.read(this, t, !1, 52, 8);
    };
    function H(r, t, e, n, a, s) {
      if (!i.isBuffer(r)) throw new TypeError('"buffer" argument must be a Buffer instance');
      if (t > a || t < s) throw new RangeError('"value" argument is out of bounds');
      if (e + n > r.length) throw new RangeError("Index out of range");
    }
    i.prototype.writeUintLE = i.prototype.writeUIntLE = function(t, e, n, a) {
      if (t = +t, e = e >>> 0, n = n >>> 0, !a) {
        const A = Math.pow(2, 8 * n) - 1;
        H(this, t, e, n, A, 0);
      }
      let s = 1, y = 0;
      for (this[e] = t & 255; ++y < n && (s *= 256); )
        this[e + y] = t / s & 255;
      return e + n;
    }, i.prototype.writeUintBE = i.prototype.writeUIntBE = function(t, e, n, a) {
      if (t = +t, e = e >>> 0, n = n >>> 0, !a) {
        const A = Math.pow(2, 8 * n) - 1;
        H(this, t, e, n, A, 0);
      }
      let s = n - 1, y = 1;
      for (this[e + s] = t & 255; --s >= 0 && (y *= 256); )
        this[e + s] = t / y & 255;
      return e + n;
    }, i.prototype.writeUint8 = i.prototype.writeUInt8 = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 1, 255, 0), this[e] = t & 255, e + 1;
    }, i.prototype.writeUint16LE = i.prototype.writeUInt16LE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 2, 65535, 0), this[e] = t & 255, this[e + 1] = t >>> 8, e + 2;
    }, i.prototype.writeUint16BE = i.prototype.writeUInt16BE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 2, 65535, 0), this[e] = t >>> 8, this[e + 1] = t & 255, e + 2;
    }, i.prototype.writeUint32LE = i.prototype.writeUInt32LE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 4, 4294967295, 0), this[e + 3] = t >>> 24, this[e + 2] = t >>> 16, this[e + 1] = t >>> 8, this[e] = t & 255, e + 4;
    }, i.prototype.writeUint32BE = i.prototype.writeUInt32BE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 4, 4294967295, 0), this[e] = t >>> 24, this[e + 1] = t >>> 16, this[e + 2] = t >>> 8, this[e + 3] = t & 255, e + 4;
    };
    function Pt(r, t, e, n, a) {
      jt(t, n, a, r, e, 7);
      let s = Number(t & BigInt(4294967295));
      r[e++] = s, s = s >> 8, r[e++] = s, s = s >> 8, r[e++] = s, s = s >> 8, r[e++] = s;
      let y = Number(t >> BigInt(32) & BigInt(4294967295));
      return r[e++] = y, y = y >> 8, r[e++] = y, y = y >> 8, r[e++] = y, y = y >> 8, r[e++] = y, e;
    }
    function Ht(r, t, e, n, a) {
      jt(t, n, a, r, e, 7);
      let s = Number(t & BigInt(4294967295));
      r[e + 7] = s, s = s >> 8, r[e + 6] = s, s = s >> 8, r[e + 5] = s, s = s >> 8, r[e + 4] = s;
      let y = Number(t >> BigInt(32) & BigInt(4294967295));
      return r[e + 3] = y, y = y >> 8, r[e + 2] = y, y = y >> 8, r[e + 1] = y, y = y >> 8, r[e] = y, e + 8;
    }
    i.prototype.writeBigUInt64LE = W(function(t, e = 0) {
      return Pt(this, t, e, BigInt(0), BigInt("0xffffffffffffffff"));
    }), i.prototype.writeBigUInt64BE = W(function(t, e = 0) {
      return Ht(this, t, e, BigInt(0), BigInt("0xffffffffffffffff"));
    }), i.prototype.writeIntLE = function(t, e, n, a) {
      if (t = +t, e = e >>> 0, !a) {
        const R = Math.pow(2, 8 * n - 1);
        H(this, t, e, n, R - 1, -R);
      }
      let s = 0, y = 1, A = 0;
      for (this[e] = t & 255; ++s < n && (y *= 256); )
        t < 0 && A === 0 && this[e + s - 1] !== 0 && (A = 1), this[e + s] = (t / y >> 0) - A & 255;
      return e + n;
    }, i.prototype.writeIntBE = function(t, e, n, a) {
      if (t = +t, e = e >>> 0, !a) {
        const R = Math.pow(2, 8 * n - 1);
        H(this, t, e, n, R - 1, -R);
      }
      let s = n - 1, y = 1, A = 0;
      for (this[e + s] = t & 255; --s >= 0 && (y *= 256); )
        t < 0 && A === 0 && this[e + s + 1] !== 0 && (A = 1), this[e + s] = (t / y >> 0) - A & 255;
      return e + n;
    }, i.prototype.writeInt8 = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 1, 127, -128), t < 0 && (t = 255 + t + 1), this[e] = t & 255, e + 1;
    }, i.prototype.writeInt16LE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 2, 32767, -32768), this[e] = t & 255, this[e + 1] = t >>> 8, e + 2;
    }, i.prototype.writeInt16BE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 2, 32767, -32768), this[e] = t >>> 8, this[e + 1] = t & 255, e + 2;
    }, i.prototype.writeInt32LE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 4, 2147483647, -2147483648), this[e] = t & 255, this[e + 1] = t >>> 8, this[e + 2] = t >>> 16, this[e + 3] = t >>> 24, e + 4;
    }, i.prototype.writeInt32BE = function(t, e, n) {
      return t = +t, e = e >>> 0, n || H(this, t, e, 4, 2147483647, -2147483648), t < 0 && (t = 4294967295 + t + 1), this[e] = t >>> 24, this[e + 1] = t >>> 16, this[e + 2] = t >>> 8, this[e + 3] = t & 255, e + 4;
    }, i.prototype.writeBigInt64LE = W(function(t, e = 0) {
      return Pt(this, t, e, -BigInt("0x8000000000000000"), BigInt("0x7fffffffffffffff"));
    }), i.prototype.writeBigInt64BE = W(function(t, e = 0) {
      return Ht(this, t, e, -BigInt("0x8000000000000000"), BigInt("0x7fffffffffffffff"));
    });
    function Ot(r, t, e, n, a, s) {
      if (e + n > r.length) throw new RangeError("Index out of range");
      if (e < 0) throw new RangeError("Index out of range");
    }
    function zt(r, t, e, n, a) {
      return t = +t, e = e >>> 0, a || Ot(r, t, e, 4), f.write(r, t, e, n, 23, 4), e + 4;
    }
    i.prototype.writeFloatLE = function(t, e, n) {
      return zt(this, t, e, !0, n);
    }, i.prototype.writeFloatBE = function(t, e, n) {
      return zt(this, t, e, !1, n);
    };
    function Wt(r, t, e, n, a) {
      return t = +t, e = e >>> 0, a || Ot(r, t, e, 8), f.write(r, t, e, n, 52, 8), e + 8;
    }
    i.prototype.writeDoubleLE = function(t, e, n) {
      return Wt(this, t, e, !0, n);
    }, i.prototype.writeDoubleBE = function(t, e, n) {
      return Wt(this, t, e, !1, n);
    }, i.prototype.copy = function(t, e, n, a) {
      if (!i.isBuffer(t)) throw new TypeError("argument should be a Buffer");
      if (n || (n = 0), !a && a !== 0 && (a = this.length), e >= t.length && (e = t.length), e || (e = 0), a > 0 && a < n && (a = n), a === n || t.length === 0 || this.length === 0) return 0;
      if (e < 0)
        throw new RangeError("targetStart out of bounds");
      if (n < 0 || n >= this.length) throw new RangeError("Index out of range");
      if (a < 0) throw new RangeError("sourceEnd out of bounds");
      a > this.length && (a = this.length), t.length - e < a - n && (a = t.length - e + n);
      const s = a - n;
      return this === t && typeof Uint8Array.prototype.copyWithin == "function" ? this.copyWithin(e, n, a) : Uint8Array.prototype.set.call(
        t,
        this.subarray(n, a),
        e
      ), s;
    }, i.prototype.fill = function(t, e, n, a) {
      if (typeof t == "string") {
        if (typeof e == "string" ? (a = e, e = 0, n = this.length) : typeof n == "string" && (a = n, n = this.length), a !== void 0 && typeof a != "string")
          throw new TypeError("encoding must be a string");
        if (typeof a == "string" && !i.isEncoding(a))
          throw new TypeError("Unknown encoding: " + a);
        if (t.length === 1) {
          const y = t.charCodeAt(0);
          (a === "utf8" && y < 128 || a === "latin1") && (t = y);
        }
      } else typeof t == "number" ? t = t & 255 : typeof t == "boolean" && (t = Number(t));
      if (e < 0 || this.length < e || this.length < n)
        throw new RangeError("Out of range index");
      if (n <= e)
        return this;
      e = e >>> 0, n = n === void 0 ? this.length : n >>> 0, t || (t = 0);
      let s;
      if (typeof t == "number")
        for (s = e; s < n; ++s)
          this[s] = t;
      else {
        const y = i.isBuffer(t) ? t : i.from(t, a), A = y.length;
        if (A === 0)
          throw new TypeError('The value "' + t + '" is invalid for argument "value"');
        for (s = 0; s < n - e; ++s)
          this[s + e] = y[s % A];
      }
      return this;
    };
    const G = {};
    function wt(r, t, e) {
      G[r] = class extends e {
        constructor() {
          super(), Object.defineProperty(this, "message", {
            value: t.apply(this, arguments),
            writable: !0,
            configurable: !0
          }), this.name = `${this.name} [${r}]`, this.stack, delete this.name;
        }
        get code() {
          return r;
        }
        set code(a) {
          Object.defineProperty(this, "code", {
            configurable: !0,
            enumerable: !0,
            value: a,
            writable: !0
          });
        }
        toString() {
          return `${this.name} [${r}]: ${this.message}`;
        }
      };
    }
    wt(
      "ERR_BUFFER_OUT_OF_BOUNDS",
      function(r) {
        return r ? `${r} is outside of buffer bounds` : "Attempt to access memory outside buffer bounds";
      },
      RangeError
    ), wt(
      "ERR_INVALID_ARG_TYPE",
      function(r, t) {
        return `The "${r}" argument must be of type number. Received type ${typeof t}`;
      },
      TypeError
    ), wt(
      "ERR_OUT_OF_RANGE",
      function(r, t, e) {
        let n = `The value of "${r}" is out of range.`, a = e;
        return Number.isInteger(e) && Math.abs(e) > 2 ** 32 ? a = qt(String(e)) : typeof e == "bigint" && (a = String(e), (e > BigInt(2) ** BigInt(32) || e < -(BigInt(2) ** BigInt(32))) && (a = qt(a)), a += "n"), n += ` It must be ${t}. Received ${a}`, n;
      },
      RangeError
    );
    function qt(r) {
      let t = "", e = r.length;
      const n = r[0] === "-" ? 1 : 0;
      for (; e >= n + 4; e -= 3)
        t = `_${r.slice(e - 3, e)}${t}`;
      return `${r.slice(0, e)}${t}`;
    }
    function ne(r, t, e) {
      K(t, "offset"), (r[t] === void 0 || r[t + e] === void 0) && Z(t, r.length - (e + 1));
    }
    function jt(r, t, e, n, a, s) {
      if (r > e || r < t) {
        const y = typeof t == "bigint" ? "n" : "";
        let A;
        throw t === 0 || t === BigInt(0) ? A = `>= 0${y} and < 2${y} ** ${(s + 1) * 8}${y}` : A = `>= -(2${y} ** ${(s + 1) * 8 - 1}${y}) and < 2 ** ${(s + 1) * 8 - 1}${y}`, new G.ERR_OUT_OF_RANGE("value", A, r);
      }
      ne(n, a, s);
    }
    function K(r, t) {
      if (typeof r != "number")
        throw new G.ERR_INVALID_ARG_TYPE(t, "number", r);
    }
    function Z(r, t, e) {
      throw Math.floor(r) !== r ? (K(r, e), new G.ERR_OUT_OF_RANGE("offset", "an integer", r)) : t < 0 ? new G.ERR_BUFFER_OUT_OF_BOUNDS() : new G.ERR_OUT_OF_RANGE(
        "offset",
        `>= 0 and <= ${t}`,
        r
      );
    }
    const ie = /[^+/0-9A-Za-z-_]/g;
    function oe(r) {
      if (r = r.split("=")[0], r = r.trim().replace(ie, ""), r.length < 2) return "";
      for (; r.length % 4 !== 0; )
        r = r + "=";
      return r;
    }
    function mt(r, t) {
      t = t || 1 / 0;
      let e;
      const n = r.length;
      let a = null;
      const s = [];
      for (let y = 0; y < n; ++y) {
        if (e = r.charCodeAt(y), e > 55295 && e < 57344) {
          if (!a) {
            if (e > 56319) {
              (t -= 3) > -1 && s.push(239, 191, 189);
              continue;
            } else if (y + 1 === n) {
              (t -= 3) > -1 && s.push(239, 191, 189);
              continue;
            }
            a = e;
            continue;
          }
          if (e < 56320) {
            (t -= 3) > -1 && s.push(239, 191, 189), a = e;
            continue;
          }
          e = (a - 55296 << 10 | e - 56320) + 65536;
        } else a && (t -= 3) > -1 && s.push(239, 191, 189);
        if (a = null, e < 128) {
          if ((t -= 1) < 0) break;
          s.push(e);
        } else if (e < 2048) {
          if ((t -= 2) < 0) break;
          s.push(
            e >> 6 | 192,
            e & 63 | 128
          );
        } else if (e < 65536) {
          if ((t -= 3) < 0) break;
          s.push(
            e >> 12 | 224,
            e >> 6 & 63 | 128,
            e & 63 | 128
          );
        } else if (e < 1114112) {
          if ((t -= 4) < 0) break;
          s.push(
            e >> 18 | 240,
            e >> 12 & 63 | 128,
            e >> 6 & 63 | 128,
            e & 63 | 128
          );
        } else
          throw new Error("Invalid code point");
      }
      return s;
    }
    function ue(r) {
      const t = [];
      for (let e = 0; e < r.length; ++e)
        t.push(r.charCodeAt(e) & 255);
      return t;
    }
    function ae(r, t) {
      let e, n, a;
      const s = [];
      for (let y = 0; y < r.length && !((t -= 2) < 0); ++y)
        e = r.charCodeAt(y), n = e >> 8, a = e % 256, s.push(a), s.push(n);
      return s;
    }
    function Gt(r) {
      return u.toByteArray(oe(r));
    }
    function lt(r, t, e, n) {
      let a;
      for (a = 0; a < n && !(a + e >= t.length || a >= r.length); ++a)
        t[a + e] = r[a];
      return a;
    }
    function O(r, t) {
      return r instanceof t || r != null && r.constructor != null && r.constructor.name != null && r.constructor.name === t.name;
    }
    function Bt(r) {
      return r !== r;
    }
    const fe = function() {
      const r = "0123456789abcdef", t = new Array(256);
      for (let e = 0; e < 16; ++e) {
        const n = e * 16;
        for (let a = 0; a < 16; ++a)
          t[n + a] = r[e] + r[a];
      }
      return t;
    }();
    function W(r) {
      return typeof BigInt > "u" ? ce : r;
    }
    function ce() {
      throw new Error("BigInt not supported");
    }
  }(de)), de;
}
var Ai = L && L.__createBinding || (Object.create ? function(o, u, f, l) {
  l === void 0 && (l = f);
  var h = Object.getOwnPropertyDescriptor(u, f);
  (!h || ("get" in h ? !u.__esModule : h.writable || h.configurable)) && (h = { enumerable: !0, get: function() {
    return u[f];
  } }), Object.defineProperty(o, l, h);
} : function(o, u, f, l) {
  l === void 0 && (l = f), o[l] = u[f];
}), Ii = L && L.__setModuleDefault || (Object.create ? function(o, u) {
  Object.defineProperty(o, "default", { enumerable: !0, value: u });
} : function(o, u) {
  o.default = u;
}), Ie = L && L.__importStar || function(o) {
  if (o && o.__esModule) return o;
  var u = {};
  if (o != null) for (var f in o) f !== "default" && Object.prototype.hasOwnProperty.call(o, f) && Ai(u, o, f);
  return Ii(u, o), u;
}, br = L && L.__awaiter || function(o, u, f, l) {
  function h(c) {
    return c instanceof f ? c : new f(function(d) {
      d(c);
    });
  }
  return new (f || (f = Promise))(function(c, d) {
    function i(x) {
      try {
        p(l.next(x));
      } catch (g) {
        d(g);
      }
    }
    function B(x) {
      try {
        p(l.throw(x));
      } catch (g) {
        d(g);
      }
    }
    function p(x) {
      x.done ? c(x.value) : h(x.value).then(i, B);
    }
    p((l = l.apply(o, u || [])).next());
  });
}, _r = L && L.__generator || function(o, u) {
  var f = { label: 0, sent: function() {
    if (c[0] & 1) throw c[1];
    return c[1];
  }, trys: [], ops: [] }, l, h, c, d;
  return d = { next: i(0), throw: i(1), return: i(2) }, typeof Symbol == "function" && (d[Symbol.iterator] = function() {
    return this;
  }), d;
  function i(p) {
    return function(x) {
      return B([p, x]);
    };
  }
  function B(p) {
    if (l) throw new TypeError("Generator is already executing.");
    for (; d && (d = 0, p[0] && (f = 0)), f; ) try {
      if (l = 1, h && (c = p[0] & 2 ? h.return : p[0] ? h.throw || ((c = h.return) && c.call(h), 0) : h.next) && !(c = c.call(h, p[1])).done) return c;
      switch (h = 0, c && (p = [p[0] & 2, c.value]), p[0]) {
        case 0:
        case 1:
          c = p;
          break;
        case 4:
          return f.label++, { value: p[1], done: !1 };
        case 5:
          f.label++, h = p[1], p = [0];
          continue;
        case 7:
          p = f.ops.pop(), f.trys.pop();
          continue;
        default:
          if (c = f.trys, !(c = c.length > 0 && c[c.length - 1]) && (p[0] === 6 || p[0] === 2)) {
            f = 0;
            continue;
          }
          if (p[0] === 3 && (!c || p[1] > c[0] && p[1] < c[3])) {
            f.label = p[1];
            break;
          }
          if (p[0] === 6 && f.label < c[1]) {
            f.label = c[1], c = p;
            break;
          }
          if (c && f.label < c[2]) {
            f.label = c[2], f.ops.push(p);
            break;
          }
          c[2] && f.ops.pop(), f.trys.pop();
          continue;
      }
      p = u.call(o, f);
    } catch (x) {
      p = [6, x], h = 0;
    } finally {
      l = c = 0;
    }
    if (p[0] & 5) throw p[1];
    return { value: p[0] ? p[1] : void 0, done: !0 };
  }
}, vr = L && L.__importDefault || function(o) {
  return o && o.__esModule ? o : { default: o };
};
Object.defineProperty(Zt, "__esModule", { value: !0 });
Zt.compute = void 0;
var Fi = Ie(X), Be = vr(Ee), Ui = vr(Or), Ci = je, $i = Ie(rr), Ri = function(o, u) {
  return u === void 0 && (u = "SHA-256"), br(void 0, void 0, void 0, function() {
    var f, l, h, c, d, i;
    return _r(this, function(B) {
      switch (B.label) {
        case 0:
          f = Fi.getCrypto(), c = !0, B.label = 1;
        case 1:
          return B.trys.push([1, 5, , 6]), f.name === "webCrypto" && typeof f.crypto.digest == "function" ? [4, f.crypto.digest(u, o)] : [3, 3];
        case 2:
          return l = B.sent(), [3, 4];
        case 3:
          f.name === "nodeCrypto" ? l = Ti(u, o, f.crypto) : c = !1, B.label = 4;
        case 4:
          return [3, 6];
        case 5:
          return d = B.sent(), d instanceof Error && (h = d.message), c = !1, [3, 6];
        case 6:
          if (c) return [3, 10];
          B.label = 7;
        case 7:
          return B.trys.push([7, 9, , 10]), [4, Mi(u, o)];
        case 8:
          return l = B.sent(), [3, 10];
        case 9:
          throw i = B.sent(), i instanceof Error ? (h = "".concat(typeof h > "u" ? "" : h, " => ").concat(i.message), new Error("UnsupportedEnvironment: ".concat(h))) : new Error("UnsupportedEnvironment");
        case 10:
          return [2, new Uint8Array(l)];
      }
    });
  });
};
Zt.compute = Ri;
var Ti = function(o, u, f) {
  var l = Be.default.hashes[o].nodeName, h = f.createHash(l);
  return h.update(u), h.digest();
}, Mi = function(o, u) {
  return br(void 0, void 0, void 0, function() {
    var f, l, h, c;
    return _r(this, function(d) {
      switch (d.label) {
        case 0:
          return o !== "MD5" ? [3, 1] : (f = (0, Ui.default)(Array.from(u), { asBytes: !0 }), [3, 4]);
        case 1:
          return ["SHA3-512", "SHA3-384", "SHA3-256", "SHA3-224"].indexOf(o) >= 0 ? (l = Be.default.hashes[o].hashSize * 8, h = new Ci.SHA3(l), [4, Promise.resolve().then(function() {
            return Ie(Si());
          })]) : [3, 3];
        case 2:
          return c = d.sent().Buffer, h.update(c.from(u)), f = h.digest("binary"), [3, 4];
        case 3:
          f = $i[Be.default.hashes[o].nodeName]().update(u).digest(), d.label = 4;
        case 4:
          return [2, new Uint8Array(f)];
      }
    });
  });
};
(function(o) {
  var u = L && L.__createBinding || (Object.create ? function(c, d, i, B) {
    B === void 0 && (B = i);
    var p = Object.getOwnPropertyDescriptor(d, i);
    (!p || ("get" in p ? !d.__esModule : p.writable || p.configurable)) && (p = { enumerable: !0, get: function() {
      return d[i];
    } }), Object.defineProperty(c, B, p);
  } : function(c, d, i, B) {
    B === void 0 && (B = i), c[B] = d[i];
  }), f = L && L.__setModuleDefault || (Object.create ? function(c, d) {
    Object.defineProperty(c, "default", { enumerable: !0, value: d });
  } : function(c, d) {
    c.default = d;
  }), l = L && L.__importStar || function(c) {
    if (c && c.__esModule) return c;
    var d = {};
    if (c != null) for (var i in c) i !== "default" && Object.prototype.hasOwnProperty.call(c, i) && u(d, c, i);
    return f(d, c), d;
  };
  Object.defineProperty(o, "__esModule", { value: !0 }), o.compute = void 0;
  var h = l(Zt);
  o.compute = h.compute, o.default = { compute: o.compute };
})(Oe);
(function(o) {
  var u = L && L.__createBinding || (Object.create ? function(w, _, b, v) {
    v === void 0 && (v = b);
    var E = Object.getOwnPropertyDescriptor(_, b);
    (!E || ("get" in E ? !_.__esModule : E.writable || E.configurable)) && (E = { enumerable: !0, get: function() {
      return _[b];
    } }), Object.defineProperty(w, v, E);
  } : function(w, _, b, v) {
    v === void 0 && (v = b), w[v] = _[b];
  }), f = L && L.__setModuleDefault || (Object.create ? function(w, _) {
    Object.defineProperty(w, "default", { enumerable: !0, value: _ });
  } : function(w, _) {
    w.default = _;
  }), l = L && L.__importStar || function(w) {
    if (w && w.__esModule) return w;
    var _ = {};
    if (w != null) for (var b in w) b !== "default" && Object.prototype.hasOwnProperty.call(w, b) && u(_, w, b);
    return f(_, w), _;
  }, h = L && L.__awaiter || function(w, _, b, v) {
    function E(S) {
      return S instanceof b ? S : new b(function(I) {
        I(S);
      });
    }
    return new (b || (b = Promise))(function(S, I) {
      function M(D) {
        try {
          F(v.next(D));
        } catch (P) {
          I(P);
        }
      }
      function U(D) {
        try {
          F(v.throw(D));
        } catch (P) {
          I(P);
        }
      }
      function F(D) {
        D.done ? S(D.value) : E(D.value).then(M, U);
      }
      F((v = v.apply(w, _ || [])).next());
    });
  }, c = L && L.__generator || function(w, _) {
    var b = { label: 0, sent: function() {
      if (S[0] & 1) throw S[1];
      return S[1];
    }, trys: [], ops: [] }, v, E, S, I;
    return I = { next: M(0), throw: M(1), return: M(2) }, typeof Symbol == "function" && (I[Symbol.iterator] = function() {
      return this;
    }), I;
    function M(F) {
      return function(D) {
        return U([F, D]);
      };
    }
    function U(F) {
      if (v) throw new TypeError("Generator is already executing.");
      for (; I && (I = 0, F[0] && (b = 0)), b; ) try {
        if (v = 1, E && (S = F[0] & 2 ? E.return : F[0] ? E.throw || ((S = E.return) && S.call(E), 0) : E.next) && !(S = S.call(E, F[1])).done) return S;
        switch (E = 0, S && (F = [F[0] & 2, S.value]), F[0]) {
          case 0:
          case 1:
            S = F;
            break;
          case 4:
            return b.label++, { value: F[1], done: !1 };
          case 5:
            b.label++, E = F[1], F = [0];
            continue;
          case 7:
            F = b.ops.pop(), b.trys.pop();
            continue;
          default:
            if (S = b.trys, !(S = S.length > 0 && S[S.length - 1]) && (F[0] === 6 || F[0] === 2)) {
              b = 0;
              continue;
            }
            if (F[0] === 3 && (!S || F[1] > S[0] && F[1] < S[3])) {
              b.label = F[1];
              break;
            }
            if (F[0] === 6 && b.label < S[1]) {
              b.label = S[1], S = F;
              break;
            }
            if (S && b.label < S[2]) {
              b.label = S[2], b.ops.push(F);
              break;
            }
            S[2] && b.ops.pop(), b.trys.pop();
            continue;
        }
        F = _.call(w, b);
      } catch (D) {
        F = [6, D], E = 0;
      } finally {
        v = S = 0;
      }
      if (F[0] & 5) throw F[1];
      return { value: F[0] ? F[1] : void 0, done: !0 };
    }
  }, d = L && L.__importDefault || function(w) {
    return w && w.__esModule ? w : { default: w };
  };
  Object.defineProperty(o, "__esModule", { value: !0 }), o.verify = o.compute = void 0;
  var i = d(be), B = l(X), p = d(Oe), x = function(w, _, b) {
    return b === void 0 && (b = "SHA-256"), h(void 0, void 0, void 0, function() {
      var v, E, S, I, M, U, F, D;
      return c(this, function(P) {
        switch (P.label) {
          case 0:
            v = B.getCrypto(), I = !0, P.label = 1;
          case 1:
            return P.trys.push([1, 6, , 7]), v.name === "webCrypto" && typeof v.crypto.importKey == "function" && typeof v.crypto.sign == "function" ? [4, v.crypto.importKey("raw", w, {
              name: "HMAC",
              hash: { name: b }
            }, !1, ["sign", "verify"])] : [3, 4];
          case 2:
            return M = P.sent(), [4, v.crypto.sign({ name: "HMAC", hash: { name: b } }, M, _)];
          case 3:
            return E = P.sent(), [3, 5];
          case 4:
            v.name === "nodeCrypto" ? (U = v.crypto.createHmac(i.default.hashes[b].nodeName, w), E = U.update(_).digest()) : I = !1, P.label = 5;
          case 5:
            return [3, 7];
          case 6:
            return F = P.sent(), F instanceof Error && (S = F.message), I = !1, [3, 7];
          case 7:
            if (I) return [3, 11];
            P.label = 8;
          case 8:
            return P.trys.push([8, 10, , 11]), [4, g(w, _, b)];
          case 9:
            return E = P.sent(), [3, 11];
          case 10:
            throw D = P.sent(), D instanceof Error ? (S = "".concat(S, " => ").concat(D.message), new Error("UnsupportedEnvironments: ".concat(S))) : new Error("UnsupportedEnvironments");
          case 11:
            return [2, new Uint8Array(E)];
        }
      });
    });
  };
  o.compute = x;
  var g = function(w, _, b) {
    return h(void 0, void 0, void 0, function() {
      var v, E, S, I, M, U, F, D;
      return c(this, function(P) {
        switch (P.label) {
          case 0:
            return v = i.default.hashes[b].blockSize, E = i.default.hashes[b].hashSize, w.length > v ? [4, p.default.compute(w, b)] : [3, 2];
          case 1:
            w = P.sent(), P.label = 2;
          case 2:
            return S = new Uint8Array(v), S.set(w), I = S.map(function(rt) {
              return 255 & (54 ^ rt);
            }), M = S.map(function(rt) {
              return 255 & (92 ^ rt);
            }), U = new Uint8Array(v + _.length), U.set(I), U.set(_, v), [4, p.default.compute(U, b)];
          case 3:
            return F = P.sent(), D = new Uint8Array(v + E), D.set(M), D.set(F, v), [2, p.default.compute(D, b)];
        }
      });
    });
  }, m = function(w, _, b, v) {
    return v === void 0 && (v = "SHA-256"), h(void 0, void 0, void 0, function() {
      var E;
      return c(this, function(S) {
        switch (S.label) {
          case 0:
            return [4, (0, o.compute)(w, _, v)];
          case 1:
            return E = S.sent(), [2, b.toString() === E.toString()];
        }
      });
    });
  };
  o.verify = m;
})(Ne);
(function(o) {
  var u = L && L.__createBinding || (Object.create ? function(c, d, i, B) {
    B === void 0 && (B = i);
    var p = Object.getOwnPropertyDescriptor(d, i);
    (!p || ("get" in p ? !d.__esModule : p.writable || p.configurable)) && (p = { enumerable: !0, get: function() {
      return d[i];
    } }), Object.defineProperty(c, B, p);
  } : function(c, d, i, B) {
    B === void 0 && (B = i), c[B] = d[i];
  }), f = L && L.__setModuleDefault || (Object.create ? function(c, d) {
    Object.defineProperty(c, "default", { enumerable: !0, value: d });
  } : function(c, d) {
    c.default = d;
  }), l = L && L.__importStar || function(c) {
    if (c && c.__esModule) return c;
    var d = {};
    if (c != null) for (var i in c) i !== "default" && Object.prototype.hasOwnProperty.call(c, i) && u(d, c, i);
    return f(d, c), d;
  };
  Object.defineProperty(o, "__esModule", { value: !0 }), o.verify = o.compute = void 0;
  var h = l(Ne);
  o.compute = h.compute, o.verify = h.verify, o.default = { compute: o.compute, verify: o.verify };
})(Le);
const Di = /* @__PURE__ */ $r(Le), Ki = {
  __name: "AuthProvider",
  props: {
    hex_hmac_signature: {
      type: String,
      required: !0
    }
  },
  setup(o) {
    const u = o, f = Ir(null), l = (p) => {
      const x = Object.keys(p).sort(), g = {};
      for (const m of x)
        g[m] = p[m];
      return g;
    }, h = (p) => [...new Uint8Array(p)].map((x) => x.toString(16).padStart(2, "0")).join(""), c = (p) => {
      const x = [];
      for (let g = 0; g < p.length; g += 2)
        x.push(Number.parseInt(p.substr(g, 2), 16));
      return new Uint8Array(x);
    };
    let d = { ...bt.initDataUnsafe };
    const i = d.hash;
    let B = "";
    return delete d.hash, d = l(d), B = Object.entries(d).map(([p, x]) => (x = typeof x == "object" ? JSON.stringify(x, null, 0) : x, `${p}=${x}`)).join(`
`).replace(/\//g, "\\/"), Di.compute(c(u.hex_hmac_signature), new TextEncoder().encode(B), "SHA-256").then((p) => {
      f.value = h(p) === i;
    }), $t("authorized", f.value), (p, x) => f.value ? Rt(p.$slots, "default", { key: 0 }) : Rt(p.$slots, "unauthorized", { key: 1 }, () => [
      x[0] || (x[0] = Fr(" Unauthorized Telegram Client "))
    ]);
  }
}, ki = ["disabled", "checked"], Yi = {
  __name: "Switch",
  props: {
    checked: {
      default: !1,
      type: Boolean
    },
    disabled: {
      default: !1,
      type: Boolean
    },
    platform: {
      type: String
    },
    colorScheme: {
      type: String
    },
    theme: {
      type: String
    }
  },
  emits: [
    "update:checked"
  ],
  setup(o, { emit: u }) {
    const f = o, l = q("platform", "android"), h = q("colorScheme", "light"), c = q("theme", "material"), d = u, i = (B) => {
      d("update:checked", B.target.checked), bt.HapticFeedback.selectionChanged();
    };
    return (B, p) => (V(), tt("div", {
      class: vt(["tele-vue-component-switch", o.disabled ? "disabled" : "", f.platform || z(l), f.colorScheme || z(h), f.theme || z(c)])
    }, [
      J("label", null, [
        J("input", {
          disabled: o.disabled,
          type: "checkbox",
          onChange: i,
          checked: o.checked
        }, null, 40, ki),
        p[0] || (p[0] = J("div", null, null, -1))
      ])
    ], 2));
  }
}, Vi = {
  __name: "List",
  props: {
    platform: {
      type: String
    },
    colorScheme: {
      type: String
    },
    theme: {
      type: String
    }
  },
  setup(o) {
    const u = o, f = q("platform", "android"), l = q("colorScheme", "light"), h = q("theme", "material");
    return (c, d) => (V(), tt("ul", {
      class: vt(["tele-vue-component-list", u.platform || z(f), u.colorScheme || z(l), u.theme || z(h)])
    }, [
      Rt(c.$slots, "default")
    ], 2));
  }
}, Li = ["disabled", "checked"], Xi = {
  __name: "Checkbox",
  props: {
    checked: {
      default: !1,
      type: Boolean
    },
    disabled: {
      default: !1,
      type: Boolean
    },
    type: {
      default: "round",
      type: String
    },
    platform: {
      type: String
    },
    colorScheme: {
      type: String
    },
    theme: {
      type: String
    }
  },
  emits: [
    "update:checked"
  ],
  setup(o, { emit: u }) {
    const f = o, l = q("platform", "android"), h = q("colorScheme", "light"), c = q("theme", "material"), d = u, i = (B) => {
      d("update:checked", B.target.checked), bt.HapticFeedback.selectionChanged();
    };
    return (B, p) => (V(), tt("div", {
      class: vt(["tele-vue-component-checkbox", o.type, o.disabled ? "disabled" : "", f.platform || z(l), f.colorScheme || z(h), f.theme || z(c)])
    }, [
      J("label", null, [
        J("input", {
          disabled: o.disabled,
          type: "checkbox",
          onChange: i,
          checked: o.checked
        }, null, 40, Li),
        p[0] || (p[0] = J("span", { class: "checkmark" }, null, -1))
      ])
    ], 2));
  }
}, Ni = { key: 0 }, Pi = { key: 0 }, Ji = {
  __name: "Section",
  props: {
    platform: {
      type: String
    },
    colorScheme: {
      type: String
    },
    theme: {
      type: String
    },
    title: {
      type: String
    },
    description: {
      type: String
    }
  },
  setup(o) {
    const u = o, f = q("platform", "android"), l = q("colorScheme", "light"), h = q("theme", "material");
    return (c, d) => (V(), tt("div", {
      class: vt(["tele-vue-component-section", u.platform || z(f), u.colorScheme || z(l), u.theme || z(h)])
    }, [
      J("section", null, [
        o.title ? (V(), tt("h3", Ni, Fe(o.title), 1)) : ye("", !0),
        J("div", null, [
          Rt(c.$slots, "default")
        ])
      ]),
      o.description ? (V(), tt("p", Pi, Fe(o.description), 1)) : ye("", !0)
    ], 2));
  }
}, Hi = ["disabled", "checked"], Zi = {
  __name: "RadioButton",
  props: {
    checked: {
      default: !1,
      type: Boolean
    },
    disabled: {
      default: !1,
      type: Boolean
    },
    platform: {
      type: String
    },
    colorScheme: {
      type: String
    },
    theme: {
      type: String
    }
  },
  emits: [
    "update:checked"
  ],
  setup(o, { emit: u }) {
    const f = o, l = q("platform", "android"), h = q("colorScheme", "light"), c = q("theme", "material"), d = u, i = (B) => {
      d("update:checked", B.target.checked), bt.HapticFeedback.selectionChanged();
    };
    return (B, p) => (V(), tt("div", {
      class: vt(["tele-vue-component-radiobutton", o.disabled ? "disabled" : "", f.platform || z(l), f.colorScheme || z(h), f.theme || z(c)])
    }, [
      J("label", null, [
        J("input", {
          disabled: o.disabled,
          type: "checkbox",
          onChange: i,
          checked: o.checked
        }, null, 40, Hi),
        p[0] || (p[0] = J("span", { class: "checkmark" }, null, -1))
      ])
    ], 2));
  }
};
export {
  Wi as AppearanceProvider,
  Ki as AuthProvider,
  Xi as Checkbox,
  Vi as List,
  qi as LocaleProvider,
  Zi as RadioButton,
  Ji as Section,
  Yi as Switch
};
